<html>
	<head>
		<meta http-equiv="content-type" content="text/html; charset=utf-8" />
		<title>Cuby 4</title>
		<link href='http://fonts.googleapis.com/css?family=Ubuntu:400,300,500,300italic,400italic,500italic' rel='stylesheet' type='text/css'>
		<link rel="stylesheet" title="Default" href="layout.css">
		<link rel="stylesheet" title="Default" href="style.css">
		<link rel="stylesheet" title="Default" href="form.css">

		<link rel="stylesheet" title="Default" href="highlight.css">
		<script src="highlight.js"></script>
		<script src="yaml.js"></script>
		<script>hljs.initHighlightingOnLoad();</script>
	</head>
	<body>
		<div id="menu">
			<a href="index.html">Introduction</a>
			<a href="authors.html">Authors</a>
			<a href="download.php">Download</a>
			<a href="installation.html">Installation</a>
			<a href="getting_started.html">Getting started</a>
			<a href="useful_tips.html">Useful tips</a>
			<span class="menusection">Reference:
			<a href="keywords.html">Keywords</a>
			<a href="protocols.html">Protocols</a>
			<a href="interfaces.html">Interfaces</a>
			<a href="methods.html">Methods</a>
			<a href="input.html">Input file</a>
			<a href="geometry.html">Geometry</a>
			<a href="atom_selection.html">Atom selection</a>
			<a href="commandline.html">Commandline</a>
			<a href="queue_systems.html">Cuby on clusters</a>
			</span>
			<hr />
		</div>
		<div id="contents">
		<div id="contents-inner">
<h1>List of all keywords</h1>
<div class="keywordblock">
<a name="alpha"></a>
<h3>
<span class="kwname">alpha:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#float">float</a></span>
<span class="kwtype"></span>
</h3>
<p>Exponent in the damping function</p>
</div>
<div class="keywordblock">
<a name="amber_amberhome"></a>
<h3>
<span class="kwname">amber_amberhome:</span>
 <span class="defvalue">"N/A"</span>
<span class="kwtype"></span>
</h3>
<p>Configuration: path to base of AMBER installation</p>
</div>
<div class="keywordblock">
<a name="amber_input_file"></a>
<h3>
<span class="kwname">amber_input_file:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>When specified, user-provided input file (for the sander module) is used,
othrevise, cuby is used to build the input.  (The input must contain the
commands to dump the results and gradient, see the input generated by
cuby for reference.)</p>
</div>
<div class="keywordblock">
<a name="amber_ld_path"></a>
<h3>
<span class="kwname">amber_ld_path:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Path to shared libraries used by amber.</p>
</div>
<div class="keywordblock">
<a name="amber_leaprc"></a>
<h3>
<span class="kwname">amber_leaprc:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#array">array</a></span>
</h3>
<p>A list (array of strings) of leaprc files loaded when prmtop is built
automatically. By default, the ff03 and gaff forcefiled are loaded from the $AMBERHOME
directory.
Each leaprc is preprocessed by cuby and following patterns are replaced with actual paths
(to allow easy construction of complex forcefields from different sources):</p>
<ul>
<li>%amberhome (path to amber installation)</li>
<li>%home (path to home directory)</li>
<li>%cuby (path to cuby inctallation)</li>
<li>%pwd (path curernt working directory)</li>
</ul>
</div>
<div class="keywordblock">
<a name="amber_sander_exe"></a>
<h3>
<span class="kwname">amber_sander_exe:</span>
 <span class="defvalue">"$AMBERHOME/bin/sander"</span>
<span class="kwtype"></span>
</h3>
<p>Configuration: path to sander executable. By default, it is taken from 
AMBERHOME directory, but this can be overriden manually.</p>
</div>
<div class="keywordblock">
<a name="amber_top_file"></a>
<h3>
<span class="kwname">amber_top_file:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>When specified, user-provided topology file is used, otherwise cuby
calls leap to generate it</p>
</div>
<div class="keywordblock">
<a name="atomic_charges"></a>
<h3>
<span class="kwname">atomic_charges:</span>
 <span class="kwallowed">
<span class="defvalue">mulliken</span> | nbo | spatial | loewdin | becke | hirshfeld | hirshfeld_fragment | forcefield
</span>
<span class="kwtype"></span>
</h3>
<p>Method of atomic charges calculation:</p>
<ul>
<li>Mulliken - Mulliken charges</li>
<li>NBO - Natural Bond Orbital analyis</li>
<li>spatial - spatial population analysis (deMon only)</li>
<li>loewdin - Loewdin population analysis (deMon only)</li>
<li>becke - Becke cell population analysis (deMon only)</li>
<li>hirshfeld - Hirshfeld population analysis using reference density of neutral atoms (deMon only)</li>
<li>hirshfeld_fragment - Hirshfeld population analysis using reference fragment density (deMon only)</li>
<li>forcefield - charges read from forcefield, MM methods</li>
</ul>
</div>
<div class="keywordblock">
<a name="atomic_charges_read"></a>
<h3>
<span class="kwname">atomic_charges_read:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>File with atomic charges for the whole system (used for electrostatic embedding).
If not provided, cuby attempts to obtain the charges from the MM interface
(might not be supported by every interface).</p>
</div>
<div class="keywordblock">
<a name="atomic_charges_write"></a>
<h3>
<span class="kwname">atomic_charges_write:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Name of the file to write the atomic charges to. If not set, the charges
are only printed but not saved to a file.</p>
</div>
<div class="keywordblock">
<a name="atomic_charges_write_pch"></a>
<h3>
<span class="kwname">atomic_charges_write_pch:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Write the charges in the format of point charges with coordinates (x,y,z,charge).
The value of the keyword should be a name of the file, nothing is written
if the keyword is not set.</p>
</div>
<div class="keywordblock">
<a name="auxiliary_basis_mp2"></a>
<h3>
<span class="kwname">auxiliary_basis_mp2:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Auxiliary basis for RI or density fitting in MP2. Default basis set is used when
this keyword is not set and the extrenal program can determine it automatically.</p>
</div>
<div class="keywordblock">
<a name="auxiliary_basis_mp2_elements"></a>
<h3>
<span class="kwname">auxiliary_basis_mp2_elements:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#hash">hash - a list of name: value pairs</a></span>
</h3>
<p>Specification of other auxiliary basis set for selected elements. The format is
a hash element: "basis_name"</p>
</div>
<div class="keywordblock">
<a name="auxiliary_basis_scf"></a>
<h3>
<span class="kwname">auxiliary_basis_scf:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Auxiliary basis for RI or density fitting in DFT/HF. Default basis set is used when
this keyword is not set and the extrenal program can determine it automatically.</p>
</div>
<div class="keywordblock">
<a name="auxiliary_basis_scf_elements"></a>
<h3>
<span class="kwname">auxiliary_basis_scf_elements:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#hash">hash - a list of name: value pairs</a></span>
</h3>
<p>Specification of other auxiliary basis set for selected elements. The format is
a hash element: "basis_name"</p>
</div>
<div class="keywordblock">
<a name="balloon_dir"></a>
<h3>
<span class="kwname">balloon_dir:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Path to the Balloon program used for building of geometries from SMILES
notation.</p>
</div>
<div class="keywordblock">
<a name="basisset"></a>
<h3>
<span class="kwname">basisset:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>AO basis set name</p>
</div>
<div class="keywordblock">
<a name="basisset_elements"></a>
<h3>
<span class="kwname">basisset_elements:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#hash">hash - a list of name: value pairs</a></span>
</h3>
<p>Specification of other basis set for selected elements. The format is
a hash element: "basis_name"</p>
</div>
<div class="keywordblock">
<a name="basisset_file"></a>
<h3>
<span class="kwname">basisset_file:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>User-provided basis set definition in a file. If not defined,
the default set of basis set of the program is used.
Following locations are searched:</p>
<ul>
<li>Calculation directory</li>
</ul>
</div>
<div class="keywordblock">
<a name="basisset_zeta"></a>
<h3>
<span class="kwname">basisset_zeta:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#integer">integer</a></span>
<span class="kwtype"></span>
</h3>
<p>Basis set cardinal number, required i.e. for extrapolations.</p>
</div>
<div class="keywordblock">
<a name="bsse_correction"></a>
<h3>
<span class="kwname">bsse_correction:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Swithes on counterpoise correction for basis set superposition error.</p>
</div>
<div class="keywordblock">
<a name="bsse_correction_3b"></a>
<h3>
<span class="kwname">bsse_correction_3b:</span>
 <span class="kwallowed">
none | <span class="defvalue">full</span>
</span>
<span class="kwtype"></span>
</h3>
<p>Counterpoise correction:</p>
<ul>
<li>none - no correction</li>
<li>full - all calculations in trimer basis</li>
</ul>
</div>
<div class="keywordblock">
<a name="calculation_print_lookup"></a>
<h3>
<span class="kwname">calculation_print_lookup:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p></p>
<ul>
<li>Result of this calculation is printed in a form useable by the lookup interface. The result contains</li>
<li>also the contribution from the addplied modifiers.</li>
</ul>
</div>
<div class="keywordblock">
<a name="cdft_constraints"></a>
<h3>
<span class="kwname">cdft_constraints:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#array">array</a></span>
</h3>
<p>Definition of the cDFT constraints, Array (each item defines one constraint)
of Hashes with following opions:</p>
<ul>
<li>type: string, either charge or spin</li>
<li>target: target value, float</li>
<li>selection: cuby selection expression</li>
<li>donor: boolean, used to label donor in coupling calculations (optional)</li>
<li>donor: acceptor, used to label acceptor in coupling calculations (optional)</li>
</ul>
</div>
<div class="keywordblock">
<a name="cdft_hda"></a>
<h3>
<span class="kwname">cdft_hda:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Enables electron transfer coupling in cDFT calculations, two constraints with flags
'donor' and 'acceptor set to true must be present</p>
</div>
<div class="keywordblock">
<a name="cfour_bin_dir"></a>
<h3>
<span class="kwname">cfour_bin_dir:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p></p>
</div>
<div class="keywordblock">
<a name="cfour_bin_dir_mpi"></a>
<h3>
<span class="kwname">cfour_bin_dir_mpi:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p></p>
</div>
<div class="keywordblock">
<a name="cfour_force_mrcc"></a>
<h3>
<span class="kwname">cfour_force_mrcc:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Use MRCC even for CCSD(T)</p>
</div>
<div class="keywordblock">
<a name="cfour_genbas_file"></a>
<h3>
<span class="kwname">cfour_genbas_file:</span>
 <span class="defvalue">"GENBAS"</span>
<span class="kwtype"></span>
</h3>
<p>Cfour basis set file. It uses ACES2 format and is usually named GENBAS.</p>
</div>
<div class="keywordblock">
<a name="cfour_keywords"></a>
<h3>
<span class="kwname">cfour_keywords:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Add extra keywords (in the CFOUR input format) to the input file</p>
</div>
<div class="keywordblock">
<a name="cfour_ld_path"></a>
<h3>
<span class="kwname">cfour_ld_path:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Path to libraries used by CFOUR, added to LD_LIBRARY_PATH before the
program is executed.</p>
</div>
<div class="keywordblock">
<a name="cfour_mrcc_bin_dir"></a>
<h3>
<span class="kwname">cfour_mrcc_bin_dir:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p></p>
</div>
<div class="keywordblock">
<a name="charge"></a>
<h3>
<span class="kwname">charge:</span>
 <span class="defvalue">0</span>
<span class="kwtype"></span>
</h3>
<p></p>
</div>
<div class="keywordblock">
<a name="charmm_exe"></a>
<h3>
<span class="kwname">charmm_exe:</span>
 <span class="defvalue">"N/A"</span>
<span class="kwtype"></span>
</h3>
<p>Path to the charmm executable</p>
</div>
<div class="keywordblock">
<a name="charmm_exe_mpi"></a>
<h3>
<span class="kwname">charmm_exe_mpi:</span>
 <span class="defvalue">"N/A"</span>
<span class="kwtype"></span>
</h3>
<p>Path to the parallel charmm executable (or a script that sets up the
MPI environment and then runs charmm)</p>
</div>
<div class="keywordblock">
<a name="charmm_extra_stream_files"></a>
<h3>
<span class="kwname">charmm_extra_stream_files:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#array">array</a></span>
</h3>
<p>Additional stream files to be applied before the calculation.</p>
</div>
<div class="keywordblock">
<a name="charmm_extras"></a>
<h3>
<span class="kwname">charmm_extras:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Extra options for charmm, included directly into the input file before
the actual calculation.</p>
</div>
<div class="keywordblock">
<a name="charmm_extras_gen"></a>
<h3>
<span class="kwname">charmm_extras_gen:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Extra options for charmm, included directly into the input file. Unlike
the charmm_extras, this keyword aplies only to the generation of the
PSF file</p>
</div>
<div class="keywordblock">
<a name="charmm_ff_par"></a>
<h3>
<span class="kwname">charmm_ff_par:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Forcefield definition file: parameters</p>
</div>
<div class="keywordblock">
<a name="charmm_ff_paths"></a>
<h3>
<span class="kwname">charmm_ff_paths:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#array">array</a></span>
</h3>
<p>Array of paths under which forcefield data (.rtf, .prm, .str) are searched if full path is not provided.</p>
</div>
<div class="keywordblock">
<a name="charmm_ff_top"></a>
<h3>
<span class="kwname">charmm_ff_top:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Forcefield definition file: topology</p>
</div>
<div class="keywordblock">
<a name="charmm_psf_file"></a>
<h3>
<span class="kwname">charmm_psf_file:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Path to a charmm PSF file prepared by charmm. The geometry used for the
calculation must be identical to the one described by the PSF file.
When this kayword is not set, charmm will attempt to generate the PSF
file automatically.</p>
</div>
<div class="keywordblock">
<a name="charmm_segment_end_patches"></a>
<h3>
<span class="kwname">charmm_segment_end_patches:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#hash">hash - a list of name: value pairs</a></span>
</h3>
<p>Pathes applied to terminal residues of each segment. The format is
a hash with entries "segment_name: start_patch, end_patch". If a segment
is not found in this keywors, autodetection of peptide chain ends is attempted
(using NTER and CTER patches), otherwise no patches are applied.</p>
</div>
<div class="keywordblock">
<a name="charmm_stream_files"></a>
<h3>
<span class="kwname">charmm_stream_files:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#array">array</a></span>
</h3>
<p>Load stream file(s) with parameters. </p>
</div>
<div class="keywordblock">
<a name="connector_orientation"></a>
<h3>
<span class="kwname">connector_orientation:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Selections (semicolon-separated) defining the coordinate frame
for the connector. Three centers (A, B, C) have to be defined for connector_type
values linear/perpendicular, one center for connector_type apex.</p>
</div>
<div class="keywordblock">
<a name="connector_type"></a>
<h3>
<span class="kwname">connector_type:</span>
 <span class="kwallowed">
<span class="defvalue">linear</span> | perpendicular | apex
</span>
<span class="kwtype"></span>
</h3>
<p>Connector orientation with respect to the molecular coordinate frame.
The linear/perpendicular connecors are defined as connected to center A,
in frame of axis AB and plane ABC provided in the connector_orientation
keyword.
The value 'apex' places the connector in the axis of all bonds the atom
is involved in, pointing outwards.</p>
</div>
<div class="keywordblock">
<a name="coordinates"></a>
<h3>
<span class="kwname">coordinates:</span>
 <span class="kwallowed">
<span class="defvalue">cartesian</span> | z-matrix | redundant
</span>
<span class="kwtype"></span>
</h3>
<p>Coordinate system used for geometry optimization. To use z-matrix,
the geometry must be loaded from a z-matrix format.</p>
</div>
<div class="keywordblock">
<a name="correlation_convergence"></a>
<h3>
<span class="kwname">correlation_convergence:</span>
 <span class="defvalue">7</span>
<span class="kwtype"></span>
</h3>
<p>Convergence threshold for iterative calculations of correlation energy, set to 10^-X a.u.</p>
</div>
<div class="keywordblock">
<a name="correlation_frozen_core"></a>
<h3>
<span class="kwname">correlation_frozen_core:</span>
 <span class="kwallowed"><span class="defvalue">yes</span> | no</span>
<span class="kwtype"></span>
</h3>
<p>Frozen core for correlation energy calculations</p>
</div>
<div class="keywordblock">
<a name="ct_scan_interval"></a>
<h3>
<span class="kwname">ct_scan_interval:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#array">array</a></span>
</h3>
<p>Interval for scanning the axis between the molecules for the best divider
plane. Format is [start, step, end]</p>
</div>
<div class="keywordblock">
<a name="cuby_local_data_dir"></a>
<h3>
<span class="kwname">cuby_local_data_dir:</span>
 <span class="defvalue">"~/.cuby4_data"</span>
<span class="kwtype"></span>
</h3>
<p>Path to directory where cuby stores local data (e.g. cache). The directory
must be writeable by user.</p>
</div>
<div class="keywordblock">
<a name="cuby_threads"></a>
<h3>
<span class="kwname">cuby_threads:</span>
 <span class="defvalue">1</span>
<span class="kwtype"></span>
</h3>
<p>Number of calculations executed by cuby in parallel.</p>
</div>
<div class="keywordblock">
<a name="d3_3body"></a>
<h3>
<span class="kwname">d3_3body:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Add 3-body term</p>
</div>
<div class="keywordblock">
<a name="d3_3body_scaling"></a>
<h3>
<span class="kwname">d3_3body_scaling:</span>
 <span class="defvalue">1.0</span>
<span class="kwtype"></span>
</h3>
<p></p>
</div>
<div class="keywordblock">
<a name="d3_3body_sr"></a>
<h3>
<span class="kwname">d3_3body_sr:</span>
 <span class="defvalue">1.333333</span>
<span class="kwtype"></span>
</h3>
<p></p>
</div>
<div class="keywordblock">
<a name="d3_a1"></a>
<h3>
<span class="kwname">d3_a1:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#float">float</a></span>
<span class="kwtype"></span>
</h3>
<p>Parameter a1 for Becke-Johnson damping function</p>
</div>
<div class="keywordblock">
<a name="d3_a2"></a>
<h3>
<span class="kwname">d3_a2:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#float">float</a></span>
<span class="kwtype"></span>
</h3>
<p>Parameter a2 for Becke-Johnson damping function</p>
</div>
<div class="keywordblock">
<a name="d3_alpha6"></a>
<h3>
<span class="kwname">d3_alpha6:</span>
 <span class="defvalue">14.0</span>
<span class="kwtype"></span>
</h3>
<p>Exponent in damping function in C6 term</p>
</div>
<div class="keywordblock">
<a name="d3_alpha8"></a>
<h3>
<span class="kwname">d3_alpha8:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#float">float</a></span>
<span class="kwtype"></span>
</h3>
<p>Exponent in damping function in C8 term. If if not setm value of d3_alpha6
+ 2 is used.</p>
</div>
<div class="keywordblock">
<a name="d3_cutoff"></a>
<h3>
<span class="kwname">d3_cutoff:</span>
 <span class="defvalue">0.0</span>
<span class="kwtype"></span>
</h3>
<p>Distance cutoff for pairwise dispersion terms. The dispersion is switched off
smoothly between this value (r) and r + d3_cutoff_buffer A. When set to 0,
no cutoff is used (default).</p>
</div>
<div class="keywordblock">
<a name="d3_cutoff_buffer"></a>
<h3>
<span class="kwname">d3_cutoff_buffer:</span>
 <span class="defvalue">2.0</span>
<span class="kwtype"></span>
</h3>
<p>Width of the region where dispersion is smoothly swithed off when cutoff is used.</p>
</div>
<div class="keywordblock">
<a name="d3_damping"></a>
<h3>
<span class="kwname">d3_damping:</span>
 <span class="kwallowed">
<span class="defvalue">bj</span> | zero | sapt | tt | none
</span>
<span class="kwtype"></span>
</h3>
<p>Selection of damping function:</p>
<ul>
<li>bj (default): the Becke-Johnson damping</li>
<li>zero: the original damping function that gets to zero at r = 0</li>
<li>tt: Tang & Toennies damping, uses the same parameters as BJ</li>
<li>sapt: damping function fitted to dispersion component from DFT-SAPT</li>
<li>none: no damping</li>
</ul>
</div>
<div class="keywordblock">
<a name="d3_data_patch"></a>
<h3>
<span class="kwname">d3_data_patch:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Patch for dispersion element data, in format identical to the main data file.</p>
</div>
<div class="keywordblock">
<a name="d3_hh_fix"></a>
<h3>
<span class="kwname">d3_hh_fix:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Hydrogen-hydrogen repulsion term for semiempirical methods. Provide parameters
using the d3_hh_para keyword.</p>
</div>
<div class="keywordblock">
<a name="d3_hh_fix_version"></a>
<h3>
<span class="kwname">d3_hh_fix_version:</span>
 <span class="defvalue">1</span>
<span class="kwtype"></span>
</h3>
<p>Value 2 switches to new, stronger repulsuion correction</p>
</div>
<div class="keywordblock">
<a name="d3_hh_para"></a>
<h3>
<span class="kwname">d3_hh_para:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#hash">hash - a list of name: value pairs</a></span>
</h3>
<p>Parameters for the H-H repulsuion correction for SQM methods. Should contain
theree parameters, k, e and r0.</p>
</div>
<div class="keywordblock">
<a name="d3_hybridization"></a>
<h3>
<span class="kwname">d3_hybridization:</span>
 <span class="kwallowed">
<span class="defvalue">grimme</span> | grimme_mod | fixed
</span>
<span class="kwtype"></span>
</h3>
<p>Hybridization used to determine atomic parameters:</p>
<ul>
<li>grimme: Grimme's continuous hybridization</li>
<li>grimme_mod: EXPERIMENTAL modification of Grimme's continuous hybridization</li>
<li>fixed: Fixed hybridization based on bond counting</li>
</ul>
</div>
<div class="keywordblock">
<a name="d3_legacy_version"></a>
<h3>
<span class="kwname">d3_legacy_version:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>In Jan 2016, we have fixed a minor bug in the interpolation of the C6 coefficients.
The results change only negligibly (0.01 kcal/mol in S66 set). This keyword enables
switching to the version prior to the bugfix.</p>
</div>
<div class="keywordblock">
<a name="d3_print"></a>
<h3>
<span class="kwname">d3_print:</span>
 <span class="kwallowed">
pairwise_energies
</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#symbol_list">(list, more items can be selected)</a></span>
<span class="kwtype"></span>
</h3>
<p></p>
</div>
<div class="keywordblock">
<a name="d3_s6"></a>
<h3>
<span class="kwname">d3_s6:</span>
 <span class="defvalue">1.0</span>
<span class="kwtype"></span>
</h3>
<p>Scaling of the C6 term</p>
</div>
<div class="keywordblock">
<a name="d3_s8"></a>
<h3>
<span class="kwname">d3_s8:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#float">float</a></span>
<span class="kwtype"></span>
</h3>
<p>Scaling of the C8 term</p>
</div>
<div class="keywordblock">
<a name="d3_scaling"></a>
<h3>
<span class="kwname">d3_scaling:</span>
 <span class="defvalue">1.0</span>
<span class="kwtype"></span>
</h3>
<p></p>
</div>
<div class="keywordblock">
<a name="d3_sr6"></a>
<h3>
<span class="kwname">d3_sr6:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#float">float</a></span>
<span class="kwtype"></span>
</h3>
<p>Radii scaling for damping function in C6 term</p>
</div>
<div class="keywordblock">
<a name="d3_sr8"></a>
<h3>
<span class="kwname">d3_sr8:</span>
 <span class="defvalue">1.0</span>
<span class="kwtype"></span>
</h3>
<p>Radii scaling for damping function in C8 term</p>
</div>
<div class="keywordblock">
<a name="dataset"></a>
<h3>
<span class="kwname">dataset:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Name of the dataset definition file</p>
</div>
<div class="keywordblock">
<a name="dataset_add_ref_to_result"></a>
<h3>
<span class="kwname">dataset_add_ref_to_result:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Data from the alternative refernce to be added to the result of calculation.</p>
</div>
<div class="keywordblock">
<a name="dataset_long_numbers"></a>
<h3>
<span class="kwname">dataset_long_numbers:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Print dataset statistics with more decimals</p>
</div>
<div class="keywordblock">
<a name="dataset_plot_columns"></a>
<h3>
<span class="kwname">dataset_plot_columns:</span>
 <span class="defvalue">4</span>
<span class="kwtype"></span>
</h3>
<p></p>
</div>
<div class="keywordblock">
<a name="dataset_plot_extra"></a>
<h3>
<span class="kwname">dataset_plot_extra:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#hash">hash - a list of name: value pairs</a></span>
</h3>
<p>Add extra series to the plots from yaml files. The value should be a hash
"name": filename.yaml</p>
</div>
<div class="keywordblock">
<a name="dataset_plot_methodname"></a>
<h3>
<span class="kwname">dataset_plot_methodname:</span>
 <span class="defvalue">"calculation"</span>
<span class="kwtype"></span>
</h3>
<p>Name of the current method to be used in plots.</p>
</div>
<div class="keywordblock">
<a name="dataset_plots_fn"></a>
<h3>
<span class="kwname">dataset_plots_fn:</span>
 <span class="defvalue">"dataset_plots"</span>
<span class="kwtype"></span>
</h3>
<p>Name of the file containing the plots, in cases all plots are written
to a single file (gnuplot_tiled and yaml format). Corresponding suffix
is added automatically.</p>
</div>
<div class="keywordblock">
<a name="dataset_print"></a>
<h3>
<span class="kwname">dataset_print:</span>
 <span class="kwallowed">
<span class="defvalue">table</span> | <span class="defvalue">stats</span> | <span class="defvalue">stats_rel</span> | <span class="defvalue">stats_groups</span>
</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#symbol_list">(list, more items can be selected)</a></span>
<span class="kwtype"></span>
</h3>
<p>List of things to be printed in the output:</p>
<ul>
<li>table - table of results for each item in the data set</li>
<li>stats - statistics, in kcal/mol</li>
<li>stats_rel - statistics, relative errors in %</li>
<li>stats_groups - brief statistics for each group of items</li>
</ul>
</div>
<div class="keywordblock">
<a name="dataset_processing"></a>
<h3>
<span class="kwname">dataset_processing:</span>
 <span class="kwallowed">
<span class="defvalue">none</span> | relative_to_average
</span>
<span class="kwtype"></span>
</h3>
<p>Processing of the results before valuation of the errors</p>
</div>
<div class="keywordblock">
<a name="dataset_reference"></a>
<h3>
<span class="kwname">dataset_reference:</span>
 <span class="defvalue">"default"</span>
<span class="kwtype"></span>
</h3>
<p>A set of reference values to be used. Some data sets might provide
alternative data, which can be selected by their name in the data
set file.</p>
</div>
<div class="keywordblock">
<a name="dataset_save_plots"></a>
<h3>
<span class="kwname">dataset_save_plots:</span>
 <span class="kwallowed">
<span class="defvalue">none</span> | text | gnuplot | gnuplot_tiled | yaml
</span>
<span class="kwtype"></span>
</h3>
<p>For dataset containg plottable series, the plots can be saved as
text files (x, y_ref, y_calc) or plotted as image using gnuplot.
The gnuplot_tiled options composes the plots into one large image
using imagemagick (named dataset_plots.png).
Number of colums is defined by keyword dataset_plot_columns.</p>
</div>
<div class="keywordblock">
<a name="dataset_select_name"></a>
<h3>
<span class="kwname">dataset_select_name:</span>
 <span class="defvalue">".*"</span>
<span class="kwtype"></span>
</h3>
<p>Selection of data set entries by name (regexp pattern)</p>
</div>
<div class="keywordblock">
<a name="dataset_select_tag"></a>
<h3>
<span class="kwname">dataset_select_tag:</span>
 <span class="defvalue">".*"</span>
<span class="kwtype"></span>
</h3>
<p>Selection of data set entries by their tags (regexp pattern)</p>
</div>
<div class="keywordblock">
<a name="dataset_short_names"></a>
<h3>
<span class="kwname">dataset_short_names:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Use short names for dataset items</p>
</div>
<div class="keywordblock">
<a name="dataset_skip_name"></a>
<h3>
<span class="kwname">dataset_skip_name:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Negated selection of data set entries by name (regexp pattern). Applied only when the keyword is used.
The entries are first selected using dataset_select_name, then those matching dataset_skip_name
are removed.</p>
</div>
<div class="keywordblock">
<a name="default_interfaces"></a>
<h3>
<span class="kwname">default_interfaces:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#hash">hash - a list of name: value pairs</a></span>
</h3>
<p>Dictionary for automated selection of interfaces for specified method</p>
</div>
<div class="keywordblock">
<a name="delete_large_files"></a>
<h3>
<span class="kwname">delete_large_files:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Delete large files created by the calculation (e.g. data needed to restart
a QM calculation) after the job finishes. The output file containing the
results is always kept.</p>
</div>
<div class="keywordblock">
<a name="demon_auxis_file"></a>
<h3>
<span class="kwname">demon_auxis_file:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Path to the deMon auxiliary basis file (AUXIS)</p>
</div>
<div class="keywordblock">
<a name="demon_basis_file"></a>
<h3>
<span class="kwname">demon_basis_file:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Path to the deMon basis file (BASIS)</p>
</div>
<div class="keywordblock">
<a name="demon_call_mpi"></a>
<h3>
<span class="kwname">demon_call_mpi:</span>
 <span class="kwallowed"><span class="defvalue">yes</span> | no</span>
<span class="kwtype"></span>
</h3>
<p>When set (default), cuby calls mpirun for parallel execution of demon. Otherwise, 
the parallel executable (or a script doing the mpi setup) is called directly
with teh number of processors as an argument.</p>
</div>
<div class="keywordblock">
<a name="demon_cdft_deformation_density"></a>
<h3>
<span class="kwname">demon_cdft_deformation_density:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>To be used with Voronoi or Becke population constraint to get the
e.g. Voronoi Deformation Density scheme.</p>
</div>
<div class="keywordblock">
<a name="demon_cdft_force_convergence"></a>
<h3>
<span class="kwname">demon_cdft_force_convergence:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p></p>
</div>
<div class="keywordblock">
<a name="demon_cdft_phase"></a>
<h3>
<span class="kwname">demon_cdft_phase:</span>
 <span class="kwallowed">
alpha | beta
</span>
<span class="kwtype"></span>
</h3>
<p>Enforce specific orbital phase convention in cDFT coupling (HDA) calculations. Applies only to development version with cDFT.</p>
</div>
<div class="keywordblock">
<a name="demon_cdft_start"></a>
<h3>
<span class="kwname">demon_cdft_start:</span>
 <span class="defvalue">0</span>
<span class="kwtype"></span>
</h3>
<p>SCF error below which cDFT is activated</p>
</div>
<div class="keywordblock">
<a name="demon_cdft_tolerance"></a>
<h3>
<span class="kwname">demon_cdft_tolerance:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#integer">integer</a></span>
<span class="kwtype"></span>
</h3>
<p>Constrained DFT tolerance</p>
</div>
<div class="keywordblock">
<a name="demon_charges_pol"></a>
<h3>
<span class="kwname">demon_charges_pol:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Add polarizability information to external charges (under development)</p>
</div>
<div class="keywordblock">
<a name="demon_constrained_dft"></a>
<h3>
<span class="kwname">demon_constrained_dft:</span>
 <span class="kwallowed">
<span class="defvalue">none</span> | mulliken | becke | loewdin | hirshfeld_fragment | hirshfeld_atomic | voronoi | spatial
</span>
<span class="kwtype"></span>
</h3>
<p>Constrained DFT (cDFT). Available population schemes are:</p>
<ul>
<li>none - no cDFT</li>
<li>spatial - uses popukation defined in -->demon_spatial_population</li>
</ul>
</div>
<div class="keywordblock">
<a name="demon_cuby_options"></a>
<h3>
<span class="kwname">demon_cuby_options:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Additional options added to the CUBY keyword when modified demon is used.</p>
</div>
<div class="keywordblock">
<a name="demon_ecps_file"></a>
<h3>
<span class="kwname">demon_ecps_file:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Path to the deMon effective core potential file (ECPS)</p>
</div>
<div class="keywordblock">
<a name="demon_exe"></a>
<h3>
<span class="kwname">demon_exe:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Path to the deMon executable</p>
</div>
<div class="keywordblock">
<a name="demon_exe_mpi"></a>
<h3>
<span class="kwname">demon_exe_mpi:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Path to the parallel (MPI) deMon executable</p>
</div>
<div class="keywordblock">
<a name="demon_ffds_file"></a>
<h3>
<span class="kwname">demon_ffds_file:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Path to the deMon force field file (FFDS)</p>
</div>
<div class="keywordblock">
<a name="demon_fragment_count"></a>
<h3>
<span class="kwname">demon_fragment_count:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#integer">integer</a></span>
<span class="kwtype"></span>
</h3>
<p>Number of fragments, required when calculations uses them</p>
</div>
<div class="keywordblock">
<a name="demon_fragment_guess"></a>
<h3>
<span class="kwname">demon_fragment_guess:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Construct initial guess from DFT calculations of fragments. The fragments must be defined in the input.</p>
</div>
<div class="keywordblock">
<a name="demon_keywords"></a>
<h3>
<span class="kwname">demon_keywords:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Additional keywords passed directly to deMon.</p>
</div>
<div class="keywordblock">
<a name="demon_lib_dir"></a>
<h3>
<span class="kwname">demon_lib_dir:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Path to a directory containing libraries needed to run the deMon executable</p>
</div>
<div class="keywordblock">
<a name="demon_maxl_auxis"></a>
<h3>
<span class="kwname">demon_maxl_auxis:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#integer">integer</a></span>
<span class="kwtype"></span>
</h3>
<p>Maximum L number of auxiliary basis functions during SCF. For deMon version
4.2.5 and above.</p>
</div>
<div class="keywordblock">
<a name="demon_mcps_file"></a>
<h3>
<span class="kwname">demon_mcps_file:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Path to the deMon model core potential file (MCPS)</p>
</div>
<div class="keywordblock">
<a name="demon_no_scf"></a>
<h3>
<span class="kwname">demon_no_scf:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>If set, energy is calculated on the initial guess of density, no SCF iterations are performed</p>
</div>
<div class="keywordblock">
<a name="demon_orbitals"></a>
<h3>
<span class="kwname">demon_orbitals:</span>
 <span class="kwallowed">
<span class="defvalue">spherical</span> | cartesian
</span>
<span class="kwtype"></span>
</h3>
<p>Orbitals used: spehrical or cartesian</p>
</div>
<div class="keywordblock">
<a name="demon_plane_axis_a"></a>
<h3>
<span class="kwname">demon_plane_axis_a:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Setup for spatial population analysis: first atom/selection defining the axis of the plane separator</p>
</div>
<div class="keywordblock">
<a name="demon_plane_axis_b"></a>
<h3>
<span class="kwname">demon_plane_axis_b:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Setup for spatial population analysis: second atom/selection defining the axis of the plane separator</p>
</div>
<div class="keywordblock">
<a name="demon_plane_ratio"></a>
<h3>
<span class="kwname">demon_plane_ratio:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#float">float</a></span>
<span class="kwtype"></span>
</h3>
<p>Setup for spatial population analysis: ratio dividing the distance between atoms A and B</p>
</div>
<div class="keywordblock">
<a name="demon_spatial_population"></a>
<h3>
<span class="kwname">demon_spatial_population:</span>
 <span class="kwallowed">
<span class="defvalue">plane_divider</span>
</span>
<span class="kwtype"></span>
</h3>
<p>Population scheme dividing the space by a plane. Three parameters should be provided:
first reference atom, second reference atom and a ratio</p>
</div>
<div class="keywordblock">
<a name="demon_version"></a>
<h3>
<span class="kwname">demon_version:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p></p>
</div>
<div class="keywordblock">
<a name="demon_with_cuby_interface"></a>
<h3>
<span class="kwname">demon_with_cuby_interface:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Set when a modified version of demon with special cuby support
is used.</p>
</div>
<div class="keywordblock">
<a name="density_convergence"></a>
<h3>
<span class="kwname">density_convergence:</span>
 <span class="defvalue">7</span>
<span class="kwtype"></span>
</h3>
<p>Density convergence threshold for scf, set to 10^-X a.u.</p>
</div>
<div class="keywordblock">
<a name="density_fitting"></a>
<h3>
<span class="kwname">density_fitting:</span>
 <span class="kwallowed">
none | scf | correlation | <span class="defvalue">both</span>
</span>
<span class="kwtype"></span>
</h3>
<p>Density fitting (or RI approximation) for SCF (DFT or HF) and correlation.
Applied only when the code and method support it. Auxiliary basis sets
must be set as needed (using auxiliary_basis_scf and auxiliary_basis_mp2).</p>
</div>
<div class="keywordblock">
<a name="development"></a>
<h3>
<span class="kwname">development:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#hash">hash - a list of name: value pairs</a></span>
</h3>
<p>This keyword is for development purposes only, it makes it possible to
temporarily add keywords without defining them explicitly.</p>
</div>
<div class="keywordblock">
<a name="dft_grid"></a>
<h3>
<span class="kwname">dft_grid:</span>
 <span class="kwallowed">
coarse | medium | <span class="defvalue">fine</span> | ultrafine | custom
</span>
<span class="kwtype"></span>
</h3>
<p>Grid for DFT calculations. The value is translated to an appropriate setup
for a specific DFT program by cuby. When value 'custom' is used, a grid
specified by keyword 'dft_grid_custom' is passed directly to the program.</p>
</div>
<div class="keywordblock">
<a name="dft_grid_custom"></a>
<h3>
<span class="kwname">dft_grid_custom:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Custom DFT grid specification in the format used by the intrfaced program.</p>
</div>
<div class="keywordblock">
<a name="dftb_3rd_order_full"></a>
<h3>
<span class="kwname">dftb_3rd_order_full:</span>
 <span class="kwallowed"><span class="defvalue">yes</span> | no</span>
<span class="kwtype"></span>
</h3>
<p>Full 3rd order calculation, when disabled, only diagonal (aka. on-site) contribution is calculated. Requires DFTB+ ver 1.2.1</p>
</div>
<div class="keywordblock">
<a name="dftb_blur_charges"></a>
<h3>
<span class="kwname">dftb_blur_charges:</span>
 <span class="defvalue">0.0</span>
<span class="kwtype"></span>
</h3>
<p>Half-width of gaussian blur of external point charges. Zero for normal
point charges.</p>
</div>
<div class="keywordblock">
<a name="dftb_cm3_charges"></a>
<h3>
<span class="kwname">dftb_cm3_charges:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Use CM3 charges in SCC. Available only in modified version of the old
DFTB program.</p>
</div>
<div class="keywordblock">
<a name="dftb_cpe"></a>
<h3>
<span class="kwname">dftb_cpe:</span>
 <span class="kwallowed">
<span class="defvalue">none</span> | cpe(q)
</span>
<span class="kwtype"></span>
</h3>
<p>CPE polarization for DFTB3, requires development version of the DFTB+ code.
The method is described in Christensen et al., J. Chem. Phys 143, 084123 (2015).</p>
</div>
<div class="keywordblock">
<a name="dftb_data_file"></a>
<h3>
<span class="kwname">dftb_data_file:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Path to a file containing data needed by some DFTB calculation (dispersion
parameters, spin constants etc.) By default, the data file from the
interfacedirectory is used, but this keyword can be used to specify another
file.</p>
</div>
<div class="keywordblock">
<a name="dftb_dispersion"></a>
<h3>
<span class="kwname">dftb_dispersion:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Original dispersion correction as implemented in DFTB+</p>
</div>
<div class="keywordblock">
<a name="dftb_do_scc"></a>
<h3>
<span class="kwname">dftb_do_scc:</span>
 <span class="kwallowed"><span class="defvalue">yes</span> | no</span>
<span class="kwtype"></span>
</h3>
<p>By default, SCC-DFTB calculation is performed. Non-SCC calculation can be invoked
by setting this keyword to false.</p>
</div>
<div class="keywordblock">
<a name="dftb_e_temp"></a>
<h3>
<span class="kwname">dftb_e_temp:</span>
 <span class="defvalue">0.0</span>
<span class="kwtype"></span>
</h3>
<p>Electronic temerature</p>
</div>
<div class="keywordblock">
<a name="dftb_exe"></a>
<h3>
<span class="kwname">dftb_exe:</span>
 <span class="defvalue">"N/A"</span>
<span class="kwtype"></span>
</h3>
<p>Configuration: path to dftb executable (original program, not DFTB+)</p>
</div>
<div class="keywordblock">
<a name="dftb_extra_input_hamiltonian"></a>
<h3>
<span class="kwname">dftb_extra_input_hamiltonian:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p></p>
</div>
<div class="keywordblock">
<a name="dftb_hubbard_derivs"></a>
<h3>
<span class="kwname">dftb_hubbard_derivs:</span>
 <span class="kwallowed">
<span class="defvalue">dftb3</span> | old | custom
</span>
<span class="kwtype"></span>
</h3>
<p>3rd order DFTB uses derivatives of Hubbard parameter. The default values are those obtained from
DFT calculation. The keyword can have following values:</p>
<ul>
<li>dftb3 - use the default parameters for DFTB3/3OB (default)</li>
<li>old - an old set of parameters used previously with the MIO set, previously default in Cuby</li>
<li>custom - use keyword dftb_hubbard_derivs_custom to provide custom values</li>
</ul>
</div>
<div class="keywordblock">
<a name="dftb_hubbard_derivs_custom"></a>
<h3>
<span class="kwname">dftb_hubbard_derivs_custom:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#hash">hash - a list of name: value pairs</a></span>
</h3>
<p>Values of custom Hubbard derivatives - see keyword dftb_hubbard_derivs.</p>
</div>
<div class="keywordblock">
<a name="dftb_safe"></a>
<h3>
<span class="kwname">dftb_safe:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Use more robust eigensolver</p>
</div>
<div class="keywordblock">
<a name="dftb_scc_convergence"></a>
<h3>
<span class="kwname">dftb_scc_convergence:</span>
 <span class="defvalue">1.0e-05</span>
<span class="kwtype"></span>
</h3>
<p>Self-consistent charges calculation convergence limit</p>
</div>
<div class="keywordblock">
<a name="dftb_slko_basepath"></a>
<h3>
<span class="kwname">dftb_slko_basepath:</span>
 <span class="defvalue">"N/A"</span>
<span class="kwtype"></span>
</h3>
<p>Configuration: path to directories containing the sets of Slater-Koster files</p>
</div>
<div class="keywordblock">
<a name="dftb_slko_format"></a>
<h3>
<span class="kwname">dftb_slko_format:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Configuration: template for Slater-Koster file naming (where X and Y are
elements, upper or lower case). If not present, name of the first file from
the slko directory is used as a template.</p>
</div>
<div class="keywordblock">
<a name="dftb_slko_set"></a>
<h3>
<span class="kwname">dftb_slko_set:</span>
 <span class="defvalue">"N/A"</span>
<span class="kwtype"></span>
</h3>
<p>Configuration: Name of the Slater-Koster parameter file set
(name of directory in dftb_slko_basepath)</p>
</div>
<div class="keywordblock">
<a name="dftb_use_dftbplus"></a>
<h3>
<span class="kwname">dftb_use_dftbplus:</span>
 <span class="kwallowed"><span class="defvalue">yes</span> | no</span>
<span class="kwtype"></span>
</h3>
<p>Use DFTB+ program (the interface supports also original DFTB)</p>
</div>
<div class="keywordblock">
<a name="dftb_xh_damping"></a>
<h3>
<span class="kwname">dftb_xh_damping:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>XH electrostatic interaction damping to improve hydrogen bonds. Requires DVTB+ ver 1.1</p>
</div>
<div class="keywordblock">
<a name="dftb_xh_damping_exp"></a>
<h3>
<span class="kwname">dftb_xh_damping_exp:</span>
 <span class="defvalue">4.95</span>
<span class="kwtype"></span>
</h3>
<p>parameter (exponent) for -->dftb_xh_damping . Default value is taken from paper on the method
optimized for use with MIO set.</p>
</div>
<div class="keywordblock">
<a name="dftbplus_exe"></a>
<h3>
<span class="kwname">dftbplus_exe:</span>
 <span class="defvalue">"N/A"</span>
<span class="kwtype"></span>
</h3>
<p>Configuration: path to dftb+ executable</p>
</div>
<div class="keywordblock">
<a name="dispersion_elements_c"></a>
<h3>
<span class="kwname">dispersion_elements_c:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#hash">hash - a list of name: value pairs</a></span>
</h3>
<p>Override c6 coefficients (unit: same as in the data file, J/nm)</p>
</div>
<div class="keywordblock">
<a name="dispersion_elements_r"></a>
<h3>
<span class="kwname">dispersion_elements_r:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#hash">hash - a list of name: value pairs</a></span>
</h3>
<p>Override element radii from the database with custom values (in Angstrom)</p>
</div>
<div class="keywordblock">
<a name="dispersion_hyb"></a>
<h3>
<span class="kwname">dispersion_hyb:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Use hybridization-dependent parameters</p>
</div>
<div class="keywordblock">
<a name="dispersion_mix"></a>
<h3>
<span class="kwname">dispersion_mix:</span>
 <span class="kwallowed">
<span class="defvalue">jurecka</span> | jurecka_safe | grimme
</span>
<span class="kwtype"></span>
</h3>
<p>Mixing of vdW eadii and C6 cooefficient in dispersion calculation. The default parameters
must be used with Jurecka's mixing. The safe version uses simpler mixing for elements
for which the effective number of valence electrons is not known (i.e. is zero).</p>
</div>
<div class="keywordblock">
<a name="dispersion_pm7_cutoff"></a>
<h3>
<span class="kwname">dispersion_pm7_cutoff:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Use a damping and cutoff for long-range contributions as used in PM7.
Enable this (along with setting the appropriate parameters) to emulate the
dispersion corerction that is built into PM7.</p>
</div>
<div class="keywordblock">
<a name="electric_field"></a>
<h3>
<span class="kwname">electric_field:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#array">array</a></span>
</h3>
<p>Homogenous external electric filed. Enetered as a cartesian vector (array), in a.u.</p>
</div>
<div class="keywordblock">
<a name="electrostatics_model"></a>
<h3>
<span class="kwname">electrostatics_model:</span>
 <span class="kwallowed">
<span class="defvalue">point_charges</span> | gaussian | slater
</span>
<span class="kwtype"></span>
</h3>
<p>Model for the electrostatic interaction. By default, point charges are used.
Diffuse electron distribution can be used as well (Gaussian or Slater monopoles),
the distribution widths (atom sizes) are calculated using Hubbard parameters
from DFTB parameter set 3OB (experimental, only some elements supported).</p>
</div>
<div class="keywordblock">
<a name="element_match_condition"></a>
<h3>
<span class="kwname">element_match_condition:</span>
 <span class="kwallowed">
<span class="defvalue">at_least</span> | exactly | no_more_than | some_from
</span>
<span class="kwtype"></span>
</h3>
<p>Condition applied to matching element list. Possible values:</p>
<ul>
<li>at_least - matches molecules containing all listed elements and possibly others</li>
<li>exactly - matches molecules consisting of all listed elements but not more</li>
<li>no_more_than - matches molecules that contain some of the listed elements but not these outside the list</li>
<li>some_from - matches molecules containg at lest one element from the list</li>
</ul>
</div>
<div class="keywordblock">
<a name="element_match_list"></a>
<h3>
<span class="kwname">element_match_list:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#symbol_list">list of symbols</a></span>
<span class="kwtype"></span>
</h3>
<p>List of elements to be matched.</p>
</div>
<div class="keywordblock">
<a name="existing_calc_dir"></a>
<h3>
<span class="kwname">existing_calc_dir:</span>
 <span class="kwallowed">
stop | recreate | <span class="defvalue">reuse</span> | read_results
</span>
<span class="kwtype"></span>
</h3>
<p>Handling of calculation directories when existing one is found:</p>
<ul>
<li>stop - program reports the existing directory and stops</li>
<li>recreate - old directory is deleted and new is generated</li>
<li>reuse - old directory is reused, new input for the external program is not generated</li>
<li>read_results - no calculation is run, old results are read</li>
</ul>
</div>
<div class="keywordblock">
<a name="explicit_correlation"></a>
<h3>
<span class="kwname">explicit_correlation:</span>
 <span class="kwallowed">
<span class="defvalue">none</span> | f12
</span>
<span class="kwtype"></span>
</h3>
<p>Explicit correlation method.</p>
</div>
<div class="keywordblock">
<a name="extension_lib_dirs"></a>
<h3>
<span class="kwname">extension_lib_dirs:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#array">array</a></span>
</h3>
<p>Paths to be passed to the compiler for building the extensions. It is
needed only when libraries from non-standard locations are used (e.g. those
built by the user in his home directory).</p>
</div>
<div class="keywordblock">
<a name="extrapolate_cbs_corr_weight"></a>
<h3>
<span class="kwname">extrapolate_cbs_corr_weight:</span>
 <span class="defvalue">1.0</span>
<span class="kwtype"></span>
</h3>
<p>Weight of the higher-order correction, 1.0 by default.</p>
</div>
<div class="keywordblock">
<a name="extrapolate_cbs_correction"></a>
<h3>
<span class="kwname">extrapolate_cbs_correction:</span>
 <span class="kwallowed"><span class="defvalue">yes</span> | no</span>
<span class="kwtype"></span>
</h3>
<p>Use higher-order correction (delta CCSD(T)) added to
MP2/CBS extrapolated correlation.</p>
</div>
<div class="keywordblock">
<a name="extrapolate_cbs_grad_mode"></a>
<h3>
<span class="kwname">extrapolate_cbs_grad_mode:</span>
 <span class="kwallowed">
<span class="defvalue">use_mp2_gradient_components</span> | separate_calculations
</span>
<span class="kwtype"></span>
</h3>
<p>Mode of gradient calculation:</p>
<ul>
<li>use_mp2_gradient_components - uses four calculations, the MP2 calcuations must yield the gradient components</li>
<li>separate_calculations - Correlation gradients calculated from separate MP2 and HF calculations</li>
</ul>
</div>
<div class="keywordblock">
<a name="fno_occ_tolerance"></a>
<h3>
<span class="kwname">fno_occ_tolerance:</span>
 <span class="defvalue">1.0e-06</span>
<span class="kwtype"></span>
</h3>
<p>Occupancy threshold for selcction of FNO orbitals. The default value is identical
to the default in Psi4.</p>
</div>
<div class="keywordblock">
<a name="forcefield_angles"></a>
<h3>
<span class="kwname">forcefield_angles:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#array">array</a></span>
</h3>
<p>Definition of forcefield terms for angles between covalent bonds</p>
</div>
<div class="keywordblock">
<a name="forcefield_bonds"></a>
<h3>
<span class="kwname">forcefield_bonds:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#array">array</a></span>
</h3>
<p>Definition of forcefield terms for covalent bonds</p>
</div>
<div class="keywordblock">
<a name="forcefield_init"></a>
<h3>
<span class="kwname">forcefield_init:</span>
 <span class="defvalue">"[]"</span>
<span class="kwtype"></span>
</h3>
<p>User ruby code run at initialization</p>
</div>
<div class="keywordblock">
<a name="forcefield_init_step"></a>
<h3>
<span class="kwname">forcefield_init_step:</span>
 <span class="defvalue">"[]"</span>
<span class="kwtype"></span>
</h3>
<p>User ruby code run before each step</p>
</div>
<div class="keywordblock">
<a name="forcefield_nonbonded"></a>
<h3>
<span class="kwname">forcefield_nonbonded:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#array">array</a></span>
</h3>
<p>Definition of forcefield terms for nonbonded pairs (all but bonds)</p>
</div>
<div class="keywordblock">
<a name="fragmentation_cut_bonds"></a>
<h3>
<span class="kwname">fragmentation_cut_bonds:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#array">array</a></span>
</h3>
<p>Bonds to be cut. Array of entries in the format 
"atom_selection; atom_selection; d_ratio",
d_ratio being the ratio between length of the bond to a hydogen cap atom
and the original bond length. The gradient is very sensitive to this,
ratio of distances optimized with the method used are needed.</p>
</div>
<div class="keywordblock">
<a name="fragmentation_dimer_file"></a>
<h3>
<span class="kwname">fragmentation_dimer_file:</span>
 <span class="defvalue">"dimers.pdb"</span>
<span class="kwtype"></span>
</h3>
<p>File to which the geometries of fragment dimers are saved. If set to empty string, no file is generated.</p>
</div>
<div class="keywordblock">
<a name="fragmentation_geometry_only"></a>
<h3>
<span class="kwname">fragmentation_geometry_only:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>If set, the program is aborted after preparation of the fragments but before
the calculations are prepared. This is useful for generating the geometries of the fragments
for the use in other calculations.</p>
</div>
<div class="keywordblock">
<a name="fragmentation_mode"></a>
<h3>
<span class="kwname">fragmentation_mode:</span>
 <span class="kwallowed">
<span class="defvalue">cut_bonds</span>
</span>
<span class="kwtype"></span>
</h3>
<p>Mode of fragmentation:</p>
<ul>
<li>cut_bonds - build the fragments by cuting user-defined bonds and all non-covalent contacts.</li>
</ul>
</div>
<div class="keywordblock">
<a name="fragmentation_monomer_file"></a>
<h3>
<span class="kwname">fragmentation_monomer_file:</span>
 <span class="defvalue">"fragments.pdb"</span>
<span class="kwtype"></span>
</h3>
<p></p>
</div>
<div class="keywordblock">
<a name="freq_mode_scan"></a>
<h3>
<span class="kwname">freq_mode_scan:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Save scans (in .xyz format) along all normal modes.</p>
</div>
<div class="keywordblock">
<a name="freq_mode_scan_batchsize"></a>
<h3>
<span class="kwname">freq_mode_scan_batchsize:</span>
 <span class="defvalue">1000</span>
<span class="kwtype"></span>
</h3>
<p>Large multidimensional scans can be split to multiple files, this number
specifies number of geometries per file.</p>
</div>
<div class="keywordblock">
<a name="freq_mode_scan_e"></a>
<h3>
<span class="kwname">freq_mode_scan_e:</span>
 <span class="defvalue">10.0</span>
<span class="kwtype"></span>
</h3>
<p>The range scanned is specified by energy (by default, in multiples of the
fundamental vibration frequency, this can be changed by keyword
freq_mode_scan_e_unit).</p>
</div>
<div class="keywordblock">
<a name="freq_mode_scan_e_unit"></a>
<h3>
<span class="kwname">freq_mode_scan_e_unit:</span>
 <span class="kwallowed">
<span class="defvalue">fundamental_frequency</span> | cm | kcal
</span>
<span class="kwtype"></span>
</h3>
<p>Energy unit for keyword req_mode_scan_e. The options are:</p>
<ul>
<li>fundamental_frequency - difefrent range for each mode, in multiples of v0</li>
<li>cm - same range for all modes, in cm^-1</li>
<li>kcal - same range for all modes, in kcal/mol</li>
</ul>
</div>
<div class="keywordblock">
<a name="freq_mode_scan_grids"></a>
<h3>
<span class="kwname">freq_mode_scan_grids:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#array">array</a></span>
</h3>
<p>Grid specification (in dimensionless normal coordinates) 
for each mode in a multidimensional scan (optional, if not set,
the energy-based displacement is used). Two possible formats are available:</p>
<ul>
<li>"seq(start, step, end)" - generates a equally-spaced grid</li>
<li>yaml array - an arbitrary list of numbers</li>
</ul>
</div>
<div class="keywordblock">
<a name="freq_mode_scan_multidimensional"></a>
<h3>
<span class="kwname">freq_mode_scan_multidimensional:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#array">array</a></span>
</h3>
<p>Multi-dimensional scan in selected modes. If the list of modes is empty,
1D scans for each mode are done instead.</p>
</div>
<div class="keywordblock">
<a name="freq_mode_scan_n"></a>
<h3>
<span class="kwname">freq_mode_scan_n:</span>
 <span class="defvalue">21</span>
<span class="kwtype"></span>
</h3>
<p>Number of points per mode.</p>
</div>
<div class="keywordblock">
<a name="freq_molden_file"></a>
<h3>
<span class="kwname">freq_molden_file:</span>
 <span class="defvalue">"vibrations.molden"</span>
<span class="kwtype"></span>
</h3>
<p>Write file for visulatiozation of the vibrations in molden.</p>
</div>
<div class="keywordblock">
<a name="freq_print"></a>
<h3>
<span class="kwname">freq_print:</span>
 <span class="kwallowed">
mode_zpve | reduced_mass | force_constants | transformation
</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#symbol_list">(list, more items can be selected)</a></span>
<span class="kwtype"></span>
</h3>
<p>List of extra features to be printed:</p>
<ul>
<li>mode_zpve - Print ZPVE (in kcal/mol) for each mode</li>
<li>reduced_mass - Print the reduced mass of each mode</li>
<li>force constants - Print the force constant of each mode</li>
<li>transformation - Transformation matrix from normal to cartesian coordinates</li>
</ul>
</div>
<div class="keywordblock">
<a name="functional"></a>
<h3>
<span class="kwname">functional:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>DFT functional, in the format recognized by cuby. It is automatically
translated to teh format used by the interfaced program. Value 'custom'
forces cuby to pass the value of the 'functional_custom' keyword to the
program without translation.</p>
</div>
<div class="keywordblock">
<a name="functional_custom"></a>
<h3>
<span class="kwname">functional_custom:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Custom DFT functional specification in the format used by the intrfaced program.</p>
</div>
<div class="keywordblock">
<a name="functional_custom_type"></a>
<h3>
<span class="kwname">functional_custom_type:</span>
 <span class="kwallowed">
lda | gga | meta_gga | hybrid | meta_hybrid | double_hybrid | range_separated
</span>
<span class="kwtype"></span>
</h3>
<p>Type of the custom functional (required in some cases where the calculation
setup depends on this). For functionals recognized by cuby, the type is
determined automatically.</p>
</div>
<div class="keywordblock">
<a name="gaussian_bin_dir"></a>
<h3>
<span class="kwname">gaussian_bin_dir:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Path to the gaussian binaries directory</p>
</div>
<div class="keywordblock">
<a name="gaussian_checkpoint"></a>
<h3>
<span class="kwname">gaussian_checkpoint:</span>
 <span class="kwallowed">
<span class="defvalue">none</span> | binary | text
</span>
<span class="kwtype"></span>
</h3>
<p>Save checkpoint file</p>
</div>
<div class="keywordblock">
<a name="gaussian_driver_exe"></a>
<h3>
<span class="kwname">gaussian_driver_exe:</span>
 <span class="defvalue">"N/A"</span>
<span class="kwtype"></span>
</h3>
<p>Path to the gaussian executable. It should be a script that sets up the
environment for gaussian and runs it in foreground. The script should
produce an output file with suffix .log from an input file (suffix .com)
passed as an argument.</p>
</div>
<div class="keywordblock">
<a name="gaussian_driver_job"></a>
<h3>
<span class="kwname">gaussian_driver_job:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Job specification, in the gaussian format (e.g. "opt" for optimization,
"freq" for frequencies etc.) This is used to construct the input.</p>
</div>
<div class="keywordblock">
<a name="gaussian_external_output"></a>
<h3>
<span class="kwname">gaussian_external_output:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>When set, cuby writes the results also in a form readable by gaussian
when it is used as a driver for external calculations
(see http://gaussian.com/g_tech/g_ur/k_external.htm). Supported by
protocols Energy, Gradient and Hessian.</p>
</div>
<div class="keywordblock">
<a name="gaussian_extra_block"></a>
<h3>
<span class="kwname">gaussian_extra_block:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Additional block (or blocks, separated by new line) of gaussian input inserted
into teh constructed input after the geometry.</p>
</div>
<div class="keywordblock">
<a name="gaussian_keywords"></a>
<h3>
<span class="kwname">gaussian_keywords:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Extra keywords in Gaussian format to be appended to the method specification line</p>
</div>
<div class="keywordblock">
<a name="gaussian_version"></a>
<h3>
<span class="kwname">gaussian_version:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Gaussian version, e.g. "g09" - the version name should correspond
to the name of the program binary.</p>
</div>
<div class="keywordblock">
<a name="geometry"></a>
<h3>
<span class="kwname">geometry:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Starting geometry. Can be specified inline in the input in cuby's format
or as a name of external file in a format recognized by cuby.</p>
</div>
<div class="keywordblock">
<a name="geometry2"></a>
<h3>
<span class="kwname">geometry2:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Secondary geometry used by some actions</p>
</div>
<div class="keywordblock">
<a name="geometry_action"></a>
<h3>
<span class="kwname">geometry_action:</span>
 <span class="kwallowed">
<span class="defvalue">none</span> | cat | partial_pdb_update | move_closest_contact_vdw_dist | selection_to_atomlist | pdb_renumber | pdb_selection_as_chain | connect | order_random | order_molecules | rmsd_fit | atomtype_bonds | build_zmat | moments_of_inertia | element_list | element_match | add_connector | orient
</span>
<span class="kwtype"></span>
</h3>
<p>Action to be performed on the geometry in addition to the standard geometry
processing:</p>
<ul>
<li>none - do nothing special</li>
<li>cat - concatenate geometry and geometry2</li>
<li>partial_pdb_update - update coordinates of matching atoms from geometry2 (atom number, atom name, residue number and residue name must be the same)</li>
<li>move_closest_contact_vdw_dist - find closest contact between two molecules and move them to vdW distance along this vector</li>
<li>selection_to_atomlist - expands a complex selection to list of atoms</li>
<li>pdb_renumber - fix numbering of atoms and residues in PDB file</li>
<li>pdb_selection_as_chain - sets chain to 'pdb_set_chain' to atoms in 'geometry_action_selection'</li>
<li>connect - connect two molecules with dummy atom connectors</li>
<li>order_random - reorders atoms randomly</li>
<li>order_molecules - reorders atoms so that individual molecules are listed one after another</li>
<li>rmsd_fit - translate/rotate geometry so that the RMSD to geometry2 is minimized</li>
<li>atomtype_bonds - prints list of all bonds, atom types involved and bond distance</li>
<li>build_zmat - build Z-matrix from cartesian coordinates</li>
<li>moments_of_inertia - print principal moments of inertia</li>
<li>element_list - print a list of elements in the molecule</li>
<li>element_match - check for presence/absence of elements, see keywords element_match_list and element_match_condition</li>
<li>add_connector - adds a connector dummy atom, see connector_type</li>
<li>orient - align selected three atoms to origin, x axis and x-y plane</li>
</ul>
</div>
<div class="keywordblock">
<a name="geometry_action_selection"></a>
<h3>
<span class="kwname">geometry_action_selection:</span>
 <span class="defvalue">"%all()"</span>
<span class="kwtype"></span>
</h3>
<p>Some actions might work on specified selections, by default, all atoms are selected</p>
</div>
<div class="keywordblock">
<a name="geometry_connect_orientation"></a>
<h3>
<span class="kwname">geometry_connect_orientation:</span>
 <span class="kwallowed">
<span class="defvalue">max_center_dist</span> | min_center_dist
</span>
<span class="kwtype"></span>
</h3>
<p>Handling of the intramolecular torsion coordinate (defined by molecule 1 center,
connection axis and molecule 2 center). Value max_center_dist maximizes the distance
of the center (trans- orientation), the other aligns the centers (cis).</p>
</div>
<div class="keywordblock">
<a name="geometry_connector"></a>
<h3>
<span class="kwname">geometry_connector:</span>
 <span class="defvalue">"auto"</span>
<span class="kwtype"></span>
</h3>
<p>Connector selection for action connect. Applied to geometry.
Value "auto" can be used to read the connector info from .xyz file comment line.</p>
</div>
<div class="keywordblock">
<a name="geometry_connector2"></a>
<h3>
<span class="kwname">geometry_connector2:</span>
 <span class="defvalue">"auto"</span>
<span class="kwtype"></span>
</h3>
<p>Connector selection for action connect. Applied to geometry2.
Value "auto" can be used to read the connector info from .xyz file comment line.</p>
</div>
<div class="keywordblock">
<a name="geometry_database_path"></a>
<h3>
<span class="kwname">geometry_database_path:</span>
 <span class="defvalue">"."</span>
<span class="kwtype"></span>
</h3>
<p>Additional path searched for geometry databases. The default location, cuby4/data/geometries,
is searched first and when the files are not found, this location is used.</p>
</div>
<div class="keywordblock">
<a name="geometry_load_charges"></a>
<h3>
<span class="kwname">geometry_load_charges:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Atomic charges to be loaded into the geometry (reads the format produced by Cuby).</p>
</div>
<div class="keywordblock">
<a name="geometry_orientation"></a>
<h3>
<span class="kwname">geometry_orientation:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Three selections (semicolon-separated) defining orientation of the molecule.
The molecule is moved so that the selected centers are aligned
to the origin, x axis and x-y plane.</p>
</div>
<div class="keywordblock">
<a name="geometry_reorder"></a>
<h3>
<span class="kwname">geometry_reorder:</span>
 <span class="kwallowed">
<span class="defvalue">original</span> | reverse | random
</span>
<span class="kwtype"></span>
</h3>
<p>Reordering of the input geometry. </p>
</div>
<div class="keywordblock">
<a name="geometry_rotate"></a>
<h3>
<span class="kwname">geometry_rotate:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>If set, the geometry is rotated along the cartesian axes. Three angles
(in degrees) should be provided, separated by commas.
Use 'rand' for a random rotation.</p>
</div>
<div class="keywordblock">
<a name="geometry_setup_from_file"></a>
<h3>
<span class="kwname">geometry_setup_from_file:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Charge and multiplicity can be read from a remark in the geometry file. For now,
only the .xyz format is supported where the second line should look like
'charge=X multiplicity=Y'. If this option is set and no data are found in the geometry,
charge 0 and multiplicity 1 are assumed.</p>
</div>
<div class="keywordblock">
<a name="geometry_template"></a>
<h3>
<span class="kwname">geometry_template:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Template geometry into which the coordinates specified by the geometry keyword are loaded.
This is an optional feature, when not set, the actual geometry is used directly.</p>
</div>
<div class="keywordblock">
<a name="geometry_update_coordinates"></a>
<h3>
<span class="kwname">geometry_update_coordinates:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>If set, the geometry read using the geometry keyword is updated with
coordinates read from this file. This feature is useful e.g. to read
coordinates into a PDB template for calculations that require
the additional information from the PDB format.</p>
</div>
<div class="keywordblock">
<a name="geometry_write"></a>
<h3>
<span class="kwname">geometry_write:</span>
 <span class="defvalue">"processed_geometry.xyz"</span>
<span class="kwtype"></span>
</h3>
<p>Name of a file to write the processed geometry to</p>
</div>
<div class="keywordblock">
<a name="geometry_write_append"></a>
<h3>
<span class="kwname">geometry_write_append:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>When set, geometry is added to the existing file</p>
</div>
<div class="keywordblock">
<a name="geometry_write_connectivity"></a>
<h3>
<span class="kwname">geometry_write_connectivity:</span>
 <span class="kwallowed">
<span class="defvalue">from_radii</span> | from_zmat
</span>
<span class="kwtype"></span>
</h3>
<p>Some geometry formats (e.g. mol2) require writing the connectivity (list of bonds).
By default, the connectivity is determined automatically from covalent radii,
optionally, bonds from z-matrix can be used if the input file was z-matrix.</p>
</div>
<div class="keywordblock">
<a name="geometry_write_format"></a>
<h3>
<span class="kwname">geometry_write_format:</span>
 <span class="kwallowed">
<span class="defvalue">auto</span> | xyz | pdb | mol2
</span>
<span class="kwtype"></span>
</h3>
<p>Geometry file format of the output. By default, it is guessed from the
file name.</p>
</div>
<div class="keywordblock">
<a name="geometry_zmat_bondlist"></a>
<h3>
<span class="kwname">geometry_zmat_bondlist:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#array">array</a></span>
</h3>
<p>This keyword allows forcing the z-matrix builder to use selected bonds.
The bond list is an array of atom selections defining the bonds - e.g. "1,2" or "3,5"</p>
</div>
<div class="keywordblock">
<a name="geometry_zmat_format"></a>
<h3>
<span class="kwname">geometry_zmat_format:</span>
 <span class="kwallowed">
<span class="defvalue">gaussian_vars</span> | gaussian | mopac
</span>
<span class="kwtype"></span>
</h3>
<p>Following z-matrix formats are availbale:</p>
<ul>
<li>gaussian_vars - Gaussian format with named variables</li>
<li>gaussian - Gaussian format with inline values</li>
<li>mopac - format for MOPAC input</li>
</ul>
</div>
<div class="keywordblock">
<a name="ghost_atoms"></a>
<h3>
<span class="kwname">ghost_atoms:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Definition of ghost atoms (nucleus without a charge, basis set functions
without electrons) by an selection expression. This is applied immediately
after the geometry is loaded, before the 'selection' keyword is used to
select some of the atoms.</p>
</div>
<div class="keywordblock">
<a name="gradient_on_point_charges"></a>
<h3>
<span class="kwname">gradient_on_point_charges:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Calculate and print also the gradient on point charges. The gradient
projection should be switched off.</p>
</div>
<div class="keywordblock">
<a name="gradient_projected"></a>
<h3>
<span class="kwname">gradient_projected:</span>
 <span class="kwallowed"><span class="defvalue">yes</span> | no</span>
<span class="kwtype"></span>
</h3>
<p>Project out translation and rotation from the gradient. Nothe that this
procedure is automatically carried out by some of the interfaces and
external programs used to calculate the gradient.</p>
</div>
<div class="keywordblock">
<a name="h_bonds4_extra_scaling"></a>
<h3>
<span class="kwname">h_bonds4_extra_scaling:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#hash">hash - a list of name: value pairs</a></span>
</h3>
<p>Manual definition of additional chrged sites to be scaled. Key of the hash is selection
of donor/acceptor atoms to be applied to, value is a sacling factor.</p>
</div>
<div class="keywordblock">
<a name="h_bonds4_parameters"></a>
<h3>
<span class="kwname">h_bonds4_parameters:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#hash">hash - a list of name: value pairs</a></span>
</h3>
<p>Parameters for the H-bond correction. By default, parameters for the selected method
are loaded automatically.</p>
</div>
<div class="keywordblock">
<a name="h_bonds4_pt_corr"></a>
<h3>
<span class="kwname">h_bonds4_pt_corr:</span>
 <span class="defvalue">0.0</span>
<span class="kwtype"></span>
</h3>
<p>Increase of proton transfer barriers</p>
</div>
<div class="keywordblock">
<a name="h_bonds4_scale_charged"></a>
<h3>
<span class="kwname">h_bonds4_scale_charged:</span>
 <span class="kwallowed"><span class="defvalue">yes</span> | no</span>
<span class="kwtype"></span>
</h3>
<p>Scaling of the correction for charged groups (COO-, NH3+).</p>
</div>
<div class="keywordblock">
<a name="h_bonds4_skip_acceptor"></a>
<h3>
<span class="kwname">h_bonds4_skip_acceptor:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#array">array</a></span>
</h3>
<p>Skip corerction for specific acceptor arrangement. Following cases are supported:</p>
<ul>
<li>OS - oxygen bound to sulfur</li>
</ul>
</div>
<div class="keywordblock">
<a name="hessian_estimate"></a>
<h3>
<span class="kwname">hessian_estimate:</span>
 <span class="kwallowed">
single_number | <span class="defvalue">by_coordinate</span> | from_lookup | fischer_almlof | lindh | swart | from_file | from_file_shift
</span>
<span class="kwtype"></span>
</h3>
<p>Estimate of the initial hessian. It applies only to
optimizations in redundant internal coordinates with
quasi-Newton optimizers. The values are:</p>
</div>
<div class="keywordblock">
<a name="hessian_projected"></a>
<h3>
<span class="kwname">hessian_projected:</span>
 <span class="kwallowed"><span class="defvalue">yes</span> | no</span>
<span class="kwtype"></span>
</h3>
<p>Project out translation and rotation from the Hessian. Nothe that this
procedure is automatically carried out by some of the interfaces and
external programs used to calculate the hessian.</p>
</div>
<div class="keywordblock">
<a name="hessian_read"></a>
<h3>
<span class="kwname">hessian_read:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>File name of the Hessian to be read (yaml format). If not specified or
empty (default value), the Hessian is calculated.</p>
</div>
<div class="keywordblock">
<a name="hessian_write"></a>
<h3>
<span class="kwname">hessian_write:</span>
 <span class="defvalue">"hessian.yaml"</span>
<span class="kwtype"></span>
</h3>
<p>File to write the calculated Hessian to. If the string is empty, Hessian is not written.</p>
</div>
<div class="keywordblock">
<a name="history_file"></a>
<h3>
<span class="kwname">history_file:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Name of the history file. If not specified, it is created from the name
of the input file if possible, or is set to "history.xyz".</p>
</div>
<div class="keywordblock">
<a name="history_freq"></a>
<h3>
<span class="kwname">history_freq:</span>
 <span class="defvalue">1</span>
<span class="kwtype"></span>
</h3>
<p>History is written each history_freq steps. If set to 0, no history is written.</p>
</div>
<div class="keywordblock">
<a name="history_selection"></a>
<h3>
<span class="kwname">history_selection:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>If set, additional history file is written for a atoms selected by this keyword.</p>
</div>
<div class="keywordblock">
<a name="hof_energy_from_vibrations"></a>
<h3>
<span class="kwname">hof_energy_from_vibrations:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>When set, the energy of the molecule is not calculated separtely but taken
from the calculation of vibrations. To get a meaningful result, the setup
in calculation_vib and calculation_ene must be identical because the
latter is still used for calculation of atom energies.</p>
</div>
<div class="keywordblock">
<a name="ignore_recommendations"></a>
<h3>
<span class="kwname">ignore_recommendations:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Can be used to prevent the program exitting when a Recommendation is
encountered. Recommendations are raised when cuby thinks the user attempts
doing something wrong but technically possible.</p>
</div>
<div class="keywordblock">
<a name="init_hessian_lookup_file"></a>
<h3>
<span class="kwname">init_hessian_lookup_file:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>File wit a lookup table used for the construction of the estimate
of the initial Hessian.</p>
</div>
<div class="keywordblock">
<a name="init_temp"></a>
<h3>
<span class="kwname">init_temp:</span>
 <span class="defvalue">10.0</span>
<span class="kwtype"></span>
</h3>
<p>Temperature used to generate initial random velocities</p>
</div>
<div class="keywordblock">
<a name="interaction_check_size"></a>
<h3>
<span class="kwname">interaction_check_size:</span>
 <span class="kwallowed"><span class="defvalue">yes</span> | no</span>
<span class="kwtype"></span>
</h3>
<p>Check for atoms missing in selections of molecule_a and molecule_b.</p>
</div>
<div class="keywordblock">
<a name="interface"></a>
<h3>
<span class="kwname">interface:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#symbol">symbol</a></span>
<span class="kwtype"></span>
</h3>
<p>Interface used for the calulation. See the list of interfaces
for possible values (online or by typing 'cuby4 interface' in shell). 
When the default value "auto" is used,
cuby will attempt to choose the interface most suitable for the
requested calculation method.</p>
</div>
<div class="keywordblock">
<a name="job"></a>
<h3>
<span class="kwname">job:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#symbol">symbol</a></span>
<span class="kwtype"></span>
</h3>
<p>Job type (calculation protocol). The available protocols are listed in the
documenation (page Protocols on the website, accessible from commandline
by running 'cuby4 protocol').</p>
</div>
<div class="keywordblock">
<a name="job_cleanup"></a>
<h3>
<span class="kwname">job_cleanup:</span>
 <span class="kwallowed"><span class="defvalue">yes</span> | no</span>
<span class="kwtype"></span>
</h3>
<p>Delete directories with external calculations when job
finishes successfuly</p>
</div>
<div class="keywordblock">
<a name="job_title"></a>
<h3>
<span class="kwname">job_title:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Custom title for the job. When not set, protocol default is used.</p>
</div>
<div class="keywordblock">
<a name="lbfgs_n"></a>
<h3>
<span class="kwname">lbfgs_n:</span>
 <span class="defvalue">5</span>
<span class="kwtype"></span>
</h3>
<p>Number of vectors stored for l-BFGS</p>
</div>
<div class="keywordblock">
<a name="linesearch"></a>
<h3>
<span class="kwname">linesearch:</span>
 <span class="kwallowed">
<span class="defvalue">none</span> | quadratic_ls | old | cubic
</span>
<span class="kwtype"></span>
</h3>
<p>Linesearch</p>
<ul>
<li>none - only step scaling</li>
<li>quadratic_ls - quadratic function approximation with weighted least squares for its derivatives</li>
<li>old - quadratic function approximation with no respoect to values, only derivatives</li>
<li>cubic - cubic function approximation</li>
</ul>
</div>
<div class="keywordblock">
<a name="maxcycles"></a>
<h3>
<span class="kwname">maxcycles:</span>
 <span class="defvalue">200</span>
<span class="kwtype"></span>
</h3>
<p>Maximum number of cycles of optimization or molecular dynamics</p>
</div>
<div class="keywordblock">
<a name="md_region"></a>
<h3>
<span class="kwname">md_region:</span>
 <span class="defvalue">"%all()"</span>
<span class="kwtype"></span>
</h3>
<p>Selection of atoms moved in the simulation. </p>
</div>
<div class="keywordblock">
<a name="measurements"></a>
<h3>
<span class="kwname">measurements:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#hash">hash - a list of name: value pairs</a></span>
</h3>
<p>A list of measuremens to be done, as a name: expression hash. For available
measuremet expressions, see the documentation of the protocol 'measure'.</p>
</div>
<div class="keywordblock">
<a name="mem"></a>
<h3>
<span class="kwname">mem:</span>
 <span class="defvalue">1000</span>
<span class="kwtype"></span>
</h3>
<p>Memory limit (in MB) passed to the external program if applicable.
It should not be set to the full amount of memory available, allow some
memory for the system. For parallel jobs, the amount of memory to be
specified depends on the parallelization mode: For shared memory
calculations, specify the total amount of memory, for MPI calculations,
memory per core should be given.</p>
</div>
<div class="keywordblock">
<a name="mem_core"></a>
<h3>
<span class="kwname">mem_core:</span>
 <span class="defvalue">75</span>
<span class="kwtype"></span>
</h3>
<p>Value in percent. Specifies how much memory is used for the user set
"core", e.g. for integral storage, in QM calculations
(Applies to Turbomole and ORCA).</p>
</div>
<div class="keywordblock">
<a name="mem_core_correlation"></a>
<h3>
<span class="kwname">mem_core_correlation:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#integer">integer</a></span>
<span class="kwtype"></span>
</h3>
<p>Value in percent. Specifies how much memory is used for the user set
"core" in corerlation energy calculation with RI. When not set,
value of keyword 'mem_core' is used instead.</p>
</div>
<div class="keywordblock">
<a name="mem_core_scf"></a>
<h3>
<span class="kwname">mem_core_scf:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#integer">integer</a></span>
<span class="kwtype"></span>
</h3>
<p>Value in percent. Specifies how much memory is used for the user set
"core" in RI-SCF calculation. When not set, value of keyword
'mem_core' is used instead.</p>
</div>
<div class="keywordblock">
<a name="method"></a>
<h3>
<span class="kwname">method:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#symbol">symbol</a></span>
<span class="kwtype"></span>
</h3>
<p>Computational method. This keyword must be specified when an external calulation
is called using an interface that implements more methods. If the interface implements
only one method, it is used by default.
The list of available methods can be found in the documentation, either online or
by command 'cuby4 method'. For example, common methods are HF, MP2, CCSD(T), forcefield, AM1...</p>
</div>
<div class="keywordblock">
<a name="mixer_weight_a"></a>
<h3>
<span class="kwname">mixer_weight_a:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#float">float</a></span>
<span class="kwtype"></span>
</h3>
<p>Weight for calculation a. The calculation should be defined in a block
calculation_a. </p>
</div>
<div class="keywordblock">
<a name="mixer_weight_b"></a>
<h3>
<span class="kwname">mixer_weight_b:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#float">float</a></span>
<span class="kwtype"></span>
</h3>
<p>Weight for calculation a. The calculation should be defined in a block
calculation_a. </p>
</div>
<div class="keywordblock">
<a name="modifiers"></a>
<h3>
<span class="kwname">modifiers:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#symbol_list">list of symbols</a></span>
<span class="kwtype"></span>
</h3>
<p>A list of interfaces applied as a modifiers to this calculations. Setup for each modifier should be provided in a block
modifier_modifiername.</p>
</div>
<div class="keywordblock">
<a name="molpro_exe"></a>
<h3>
<span class="kwname">molpro_exe:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Path to the molpro binary</p>
</div>
<div class="keywordblock">
<a name="molpro_lib_dir"></a>
<h3>
<span class="kwname">molpro_lib_dir:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Path to the directory containing libraries needed by Molpro (set using
the -L commandline option).</p>
</div>
<div class="keywordblock">
<a name="mopac_corrections"></a>
<h3>
<span class="kwname">mopac_corrections:</span>
 <span class="kwallowed">
<span class="defvalue">none</span> | d2 | h2 | dh2 | dh+ | d3 | h4 | d3h4 | d3h4x
</span>
<span class="kwtype"></span>
</h3>
<p>Corrections for noncovalent interactions for PM6 as implemented in MOPAC.</p>
</div>
<div class="keywordblock">
<a name="mopac_exe"></a>
<h3>
<span class="kwname">mopac_exe:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>MOPAC executable</p>
</div>
<div class="keywordblock">
<a name="mopac_keywords"></a>
<h3>
<span class="kwname">mopac_keywords:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Extra keywords (in the MOPAC format) added into the input</p>
</div>
<div class="keywordblock">
<a name="mopac_mozyme"></a>
<h3>
<span class="kwname">mopac_mozyme:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Use of the linear scaling MOZYME algorithm. MOPAC documentation:
http://openmopac.net/manual/mozyme.html</p>
</div>
<div class="keywordblock">
<a name="mopac_parallel"></a>
<h3>
<span class="kwname">mopac_parallel:</span>
 <span class="kwallowed"><span class="defvalue">yes</span> | no</span>
<span class="kwtype"></span>
</h3>
<p>The latest version allows parralelization, but the input can not be
used with older versions of MOPAC2012. Disable this option when
older version is used.</p>
</div>
<div class="keywordblock">
<a name="mopac_peptide_bond_fix"></a>
<h3>
<span class="kwname">mopac_peptide_bond_fix:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Controls the corerction for peptide bond torsion. It was introduced as an
addition to the RM1, PM6 and PM7 methods, therefore it is switched off by
default for backwards compatibility. MOPAC documentation:
http://openmopac.net/manual/mmok.html</p>
</div>
<div class="keywordblock">
<a name="mopac_precise"></a>
<h3>
<span class="kwname">mopac_precise:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Use tight thresholds, needed for accurate gradient. MOPAC documentation:
http://openmopac.net/manual/precise.html</p>
</div>
<div class="keywordblock">
<a name="mopac_setcharge"></a>
<h3>
<span class="kwname">mopac_setcharge:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#hash">hash - a list of name: value pairs</a></span>
</h3>
<p>Manual specification atomic charges for MOZYME, hash of charges indexed
by atom selection. The charge can be '0', '+' or '-'. Described at
http://openmopac.net/manual/Labels.html</p>
</div>
<div class="keywordblock">
<a name="mopac_setpi"></a>
<h3>
<span class="kwname">mopac_setpi:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#array">array</a></span>
</h3>
<p>Manual specification of double bonds for MOZYME, array of atom pairs
(defined by the selection expression, e.g. as atom numbers separated by
comma). Described at http://openmopac.net/manual/setpi.html</p>
</div>
<div class="keywordblock">
<a name="mopac_version"></a>
<h3>
<span class="kwname">mopac_version:</span>
 <span class="kwallowed">
<span class="defvalue">mopac2012</span> | mopac7
</span>
<span class="kwtype"></span>
</h3>
<p>Version of MOPAC to be used. By default, MOPAC 2012 is used, it is however
possible to switch to the legacy MOPAC 7 but the functionality is limited.</p>
</div>
<div class="keywordblock">
<a name="mrcc_bin_dir"></a>
<h3>
<span class="kwname">mrcc_bin_dir:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Path to the directory with mrcc executables</p>
</div>
<div class="keywordblock">
<a name="mrcc_lno_domrad"></a>
<h3>
<span class="kwname">mrcc_lno_domrad:</span>
 <span class="defvalue">10</span>
<span class="kwtype"></span>
</h3>
<p>Domain radius for LNO-CC calculations. See the original paper for details.</p>
</div>
<div class="keywordblock">
<a name="mrcc_lno_eps_o"></a>
<h3>
<span class="kwname">mrcc_lno_eps_o:</span>
 <span class="defvalue">5</span>
<span class="kwtype"></span>
</h3>
<p>Threshold for LNO-CC calculations, as 10^X, the default value is rather tight. See the original paper for details.</p>
</div>
<div class="keywordblock">
<a name="mrcc_lno_eps_v"></a>
<h3>
<span class="kwname">mrcc_lno_eps_v:</span>
 <span class="defvalue">6</span>
<span class="kwtype"></span>
</h3>
<p>Threshold for LNO-CC calculations, as 10^X, the default value is rather tight. See the original paper for details.</p>
</div>
<div class="keywordblock">
<a name="mrcc_local_cc"></a>
<h3>
<span class="kwname">mrcc_local_cc:</span>
 <span class="kwallowed">
<span class="defvalue">none</span> | lno
</span>
<span class="kwtype"></span>
</h3>
<p>Switches on local coupled cluster method:</p>
<ul>
<li>lno: Local natural orbitals, http://dx.doi.org/10.1063/1.4819401</li>
</ul>
</div>
<div class="keywordblock">
<a name="multiplicity"></a>
<h3>
<span class="kwname">multiplicity:</span>
 <span class="defvalue">1</span>
<span class="kwtype"></span>
</h3>
<p></p>
</div>
<div class="keywordblock">
<a name="multistep_print"></a>
<h3>
<span class="kwname">multistep_print:</span>
 <span class="kwallowed"><span class="defvalue">yes</span> | no</span>
<span class="kwtype"></span>
</h3>
<p>Controls printing of step headers and output of each step</p>
</div>
<div class="keywordblock">
<a name="multistep_result_eval"></a>
<h3>
<span class="kwname">multistep_result_eval:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Alternatively to multistep_result_expression, a piece of ruby code can be
called on the results. For experts only.</p>
</div>
<div class="keywordblock">
<a name="multistep_result_expression"></a>
<h3>
<span class="kwname">multistep_result_expression:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>The result of a multistep job is evauated using this expression after all
the steps have been completed. For experts only, this requires calling
internal data structures of the multistep job.</p>
</div>
<div class="keywordblock">
<a name="multistep_result_name"></a>
<h3>
<span class="kwname">multistep_result_name:</span>
 <span class="defvalue">"Result expression value"</span>
<span class="kwtype"></span>
</h3>
<p>Name of the result calculated using multistep_result_expression keyword</p>
</div>
<div class="keywordblock">
<a name="neb_fconst"></a>
<h3>
<span class="kwname">neb_fconst:</span>
 <span class="defvalue">20.0</span>
<span class="kwtype"></span>
</h3>
<p>Force constant of the harmonic potential connecting the points on the
elastic band (kcal/mol/A).</p>
</div>
<div class="keywordblock">
<a name="neb_mep_file"></a>
<h3>
<span class="kwname">neb_mep_file:</span>
 <span class="defvalue">"minimum_energy_path.xyz"</span>
<span class="kwtype"></span>
</h3>
<p>The file to which the minimum energy path is written to. It is updated in each step.</p>
</div>
<div class="keywordblock">
<a name="neb_optimize_endpoints"></a>
<h3>
<span class="kwname">neb_optimize_endpoints:</span>
 <span class="kwallowed"><span class="defvalue">yes</span> | no</span>
<span class="kwtype"></span>
</h3>
<p>Optimization of the end points of the MEP can be disabled if the provided
geometries are already optimized.</p>
</div>
<div class="keywordblock">
<a name="neb_version"></a>
<h3>
<span class="kwname">neb_version:</span>
 <span class="kwallowed">
<span class="defvalue">improved</span> | original
</span>
<span class="kwtype"></span>
</h3>
<p>Version of the formulation of the NEB, see the references in the documentation.</p>
</div>
<div class="keywordblock">
<a name="numerical_gradient_atomic_charges"></a>
<h3>
<span class="kwname">numerical_gradient_atomic_charges:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Optional calculation of the derivatives of the atomic charges.</p>
</div>
<div class="keywordblock">
<a name="numerical_gradient_step"></a>
<h3>
<span class="kwname">numerical_gradient_step:</span>
 <span class="defvalue">0.0025</span>
<span class="kwtype"></span>
</h3>
<p></p>
</div>
<div class="keywordblock">
<a name="numerical_gradient_type"></a>
<h3>
<span class="kwname">numerical_gradient_type:</span>
 <span class="kwallowed">
single_sided | <span class="defvalue">central</span> | tetrahedral
</span>
<span class="kwtype"></span>
</h3>
<p>Number of points per atom used for numerical differentiation:</p>
<ul>
<li>single_sided - 3 points, least robust</li>
<li>central - 6 points, most accurate</li>
<li>tetrahedral - 4 points</li>
</ul>
</div>
<div class="keywordblock">
<a name="numerical_hessian_e_order"></a>
<h3>
<span class="kwname">numerical_hessian_e_order:</span>
 <span class="defvalue">2</span>
<span class="kwtype"></span>
</h3>
<p>Hessian from energy: Order of the formula (2 or 4)</p>
</div>
<div class="keywordblock">
<a name="numerical_hessian_step"></a>
<h3>
<span class="kwname">numerical_hessian_step:</span>
 <span class="defvalue">0.001</span>
<span class="kwtype"></span>
</h3>
<p>The size of the displacemeni (in Angstrom) used in the numerical
differentiation.</p>
</div>
<div class="keywordblock">
<a name="numerical_hessian_type"></a>
<h3>
<span class="kwname">numerical_hessian_type:</span>
 <span class="kwallowed">
single_sided | <span class="defvalue">central</span> | tetrahedral | energy_only
</span>
<span class="kwtype"></span>
</h3>
<p>Number of points per atom used for numerical differentiation:</p>
<ul>
<li>single_sided - 3 points, least robust</li>
<li>central - 6 points, most accurate</li>
<li>tetrahedral - 4 points</li>
<li>energy_only - parabolic fit from energies (the other methods use gradient)</li>
</ul>
</div>
<div class="keywordblock">
<a name="opt_auto_restart"></a>
<h3>
<span class="kwname">opt_auto_restart:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>When enabled, restart file 'auto_restart.xyz' is written in each step, and
if this file is found at the beginning of an optimization, the geometry
is updated with coordinates from this file.</p>
</div>
<div class="keywordblock">
<a name="opt_convlimit_e"></a>
<h3>
<span class="kwname">opt_convlimit_e:</span>
 <span class="defvalue">0.006</span>
<span class="kwtype"></span>
</h3>
<p>Convergence limit for energy change between steps (kcal/mol).
If set to 0, this criterium is not evaluated.</p>
</div>
<div class="keywordblock">
<a name="opt_convlimit_max_g"></a>
<h3>
<span class="kwname">opt_convlimit_max_g:</span>
 <span class="defvalue">1.2</span>
<span class="kwtype"></span>
</h3>
<p>Convergence limit for maximum gradient component (kcal/mol/A).
If set to 0, this criterium is not evaluated.</p>
</div>
<div class="keywordblock">
<a name="opt_convlimit_rms_g"></a>
<h3>
<span class="kwname">opt_convlimit_rms_g:</span>
 <span class="defvalue">0.6</span>
<span class="kwtype"></span>
</h3>
<p>Convergence limit for root mean square of gradient (gradient norm) (kcal/mol/A).
If set to 0, this criterium is not evaluated.</p>
</div>
<div class="keywordblock">
<a name="opt_diagonal_h0"></a>
<h3>
<span class="kwname">opt_diagonal_h0:</span>
 <span class="defvalue">1000</span>
<span class="kwtype"></span>
</h3>
<p>Initial value of diagonal elements of the hessian</p>
</div>
<div class="keywordblock">
<a name="opt_freeze_cartesian"></a>
<h3>
<span class="kwname">opt_freeze_cartesian:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#hash">hash - a list of name: value pairs</a></span>
</h3>
<p>Allows freezing some cartesian coordinates. The input should be a hash in
a form "atom_index: xyz" (only some coordinates can be selected, e.g. by
"x" or "xz")</p>
</div>
<div class="keywordblock">
<a name="opt_qn_update_formula"></a>
<h3>
<span class="kwname">opt_qn_update_formula:</span>
 <span class="kwallowed">
<span class="defvalue">bfgs</span> | dfp | broyden | sr1 | dfp_bfgs | sr1_bfgs
</span>
<span class="kwtype"></span>
</h3>
<p>Hessian update formula for quasi-Newton optimizers</p>
</div>
<div class="keywordblock">
<a name="opt_quality"></a>
<h3>
<span class="kwname">opt_quality:</span>
 <span class="defvalue">1.0</span>
<span class="kwtype"></span>
</h3>
<p>One-number control of optimization quality - multiplier of convergence limits. The default
value sets rather loose thresholds suitable for optimization of rigid molecules only.
When accurate geometries of flexible molecules or noncovalent complexes are needed,
opt_quality should be set to 0.1 or lower.</p>
</div>
<div class="keywordblock">
<a name="opt_refuse_steps"></a>
<h3>
<span class="kwname">opt_refuse_steps:</span>
 <span class="kwallowed"><span class="defvalue">yes</span> | no</span>
<span class="kwtype"></span>
</h3>
<p>Refuse bad optimization steps and try shorter one</p>
</div>
<div class="keywordblock">
<a name="opt_tr_mode"></a>
<h3>
<span class="kwname">opt_tr_mode:</span>
 <span class="kwallowed">
abs | <span class="defvalue">max</span> | components
</span>
<span class="kwtype"></span>
</h3>
<p>Step size scaling mode for RFO, QN and LBFGS (TRIM uses another means of step size control):</p>
<ul>
<li>abs - Step is scaled so that its absolute value is not larger than the limit</li>
<li>components - Each component of the step vector is scaled separately</li>
<li>max - Step is scaled so that the largest component is not larger than the limit</li>
</ul>
</div>
<div class="keywordblock">
<a name="opt_tr_update"></a>
<h3>
<span class="kwname">opt_tr_update:</span>
 <span class="kwallowed"><span class="defvalue">yes</span> | no</span>
<span class="kwtype"></span>
</h3>
<p>Update of trust radius during the optimization</p>
</div>
<div class="keywordblock">
<a name="optimize_print"></a>
<h3>
<span class="kwname">optimize_print:</span>
 <span class="kwallowed">
<span class="defvalue">steps</span> | steps_as_dots | final_energy | statistics
</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#symbol_list">(list, more items can be selected)</a></span>
<span class="kwtype"></span>
</h3>
<p>Optional printing:</p>
<ul>
<li>steps - print complete information in each step</li>
<li>steps_as_dots - one character per step, dot or 'o' and 'O' as more criteria are fulfilled</li>
<li>final_energy - print final energy after optimization</li>
<li>statistics - print counts of various events encountered by the optimizer</li>
</ul>
</div>
<div class="keywordblock">
<a name="optimize_region"></a>
<h3>
<span class="kwname">optimize_region:</span>
 <span class="defvalue">"%all()"</span>
<span class="kwtype"></span>
</h3>
<p>Selection of atoms to be optimized.</p>
</div>
<div class="keywordblock">
<a name="optimizer"></a>
<h3>
<span class="kwname">optimizer:</span>
 <span class="kwallowed">
<span class="defvalue">auto</span> | rfo | trim | lbfgs | plbfgs | quasi_newton | nelder_mead
</span>
<span class="kwtype"></span>
</h3>
<p>Optimization algorithm. The default option 'auto' switches between
quasi-Newton (BFGS) for small (less than 100 atoms) and LBFGS optimizers.
The Nelder-Mead simplex method is not intended for geometry optimization
but for other applications.</p>
</div>
<div class="keywordblock">
<a name="orca_bin_dir"></a>
<h3>
<span class="kwname">orca_bin_dir:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Path to the directory containing the orca binaries</p>
</div>
<div class="keywordblock">
<a name="orca_dlpno"></a>
<h3>
<span class="kwname">orca_dlpno:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Enable the Domain-localized pair natural orbitals approximation in
CCSD and CCSD(T) calculation. </p>
</div>
<div class="keywordblock">
<a name="orca_ecp"></a>
<h3>
<span class="kwname">orca_ecp:</span>
 <span class="kwallowed">
<span class="defvalue">none</span> | auto
</span>
<span class="kwtype"></span>
</h3>
<p>Pseudopotentials. When set to 'auto', ORCA assigns pseudopotentials
automatically fot the selected basis sets.</p>
</div>
<div class="keywordblock">
<a name="orca_extra_input"></a>
<h3>
<span class="kwname">orca_extra_input:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p></p>
</div>
<div class="keywordblock">
<a name="orca_keywords"></a>
<h3>
<span class="kwname">orca_keywords:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Additional keywords passed to ORCA, in the simple format (a line beginning with !)</p>
</div>
<div class="keywordblock">
<a name="orca_mpi_setup"></a>
<h3>
<span class="kwname">orca_mpi_setup:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>A shell command(s) for loading the MPI environment (parallel-calculations only)</p>
</div>
<div class="keywordblock">
<a name="parallel"></a>
<h3>
<span class="kwname">parallel:</span>
 <span class="defvalue">1</span>
<span class="kwtype"></span>
</h3>
<p>Number of CPUs used by the external program running in parallel mode.</p>
</div>
<div class="keywordblock">
<a name="parallel_mode"></a>
<h3>
<span class="kwname">parallel_mode:</span>
 <span class="kwallowed">
<span class="defvalue">default</span> | mpi | shm | ga
</span>
<span class="kwtype"></span>
</h3>
<p>Mode of parallel execution of the external software. Use the value 'default'
unless the external program supports multiple options. The options are:</p>
<ul>
<li>MPI - message passing interface (note that each CPU uses the memory provided by mem keyword)</li>
<li>SHM - shared memory, e.g. multi-threaded parallelism</li>
<li>GA - global arrays - MPI parallelization with shared memory</li>
</ul>
</div>
<div class="keywordblock">
<a name="pdb_extra_columns"></a>
<h3>
<span class="kwname">pdb_extra_columns:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Controls printing of additional columns in PDB files (beyond coordinates).
Applicable to all tasks where PDB file is written.</p>
</div>
<div class="keywordblock">
<a name="pdb_set_chain"></a>
<h3>
<span class="kwname">pdb_set_chain:</span>
 <span class="defvalue">"X"</span>
<span class="kwtype"></span>
</h3>
<p>PDB chain value for action pdb_selection_as_chain     </p>
</div>
<div class="keywordblock">
<a name="plbfgs_colors"></a>
<h3>
<span class="kwname">plbfgs_colors:</span>
 <span class="defvalue">1</span>
<span class="kwtype"></span>
</h3>
<p>How many colors to to calculate in one step. The number of calculations is then
1 + 3 * plbfgs_colors. The most efficient setup is to keep the number of colors
1 and set cuby_threads to 4 to perform all the four calculations in parallel.</p>
</div>
<div class="keywordblock">
<a name="plbfgs_init_d"></a>
<h3>
<span class="kwname">plbfgs_init_d:</span>
 <span class="defvalue">1.0e-05</span>
<span class="kwtype"></span>
</h3>
<p>init. value of d for num. differentiation</p>
</div>
<div class="keywordblock">
<a name="point_charges"></a>
<h3>
<span class="kwname">point_charges:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>A list of point charges defining an extarnal electrostatic field. One
charge per line, in format x y z charge. (Coordinates in Angstrom, charge
in elemental charge units). Note that only some interfaces support
the use of point charges.</p>
</div>
<div class="keywordblock">
<a name="prepare_only"></a>
<h3>
<span class="kwname">prepare_only:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>When enabled, cuby prepares the calculations and exits before
running them.</p>
</div>
<div class="keywordblock">
<a name="print"></a>
<h3>
<span class="kwname">print:</span>
 <span class="kwallowed">
timing | energy_decomposition | subsystem_results | dipole | polarizability | input_geometry | step_timing | gradient | molecular_orbitals
</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#symbol_list">(list, more items can be selected)</a></span>
<span class="kwtype"></span>
</h3>
<p>List of extra features to be printed:</p>
<ul>
<li>timing: print time spent in the program (global)</li>
<li>energy_decomposition: components of the calculated energy, if they are present (global)</li>
<li>subsystem_results: print results for subsystems (protocol interaction, threebody)</li>
<li>dipole: dipole moment, if available</li>
<li>polarizability: molecular polarizability, if available</li>
<li>input_geometry: print the geometry used at the beginning of the calculation (global)</li>
<li>step_timing: print timing for individual steps of the calculation (protocol optimize)</li>
<li>gradient: print gradient used in the optimization (protocol parametrize)</li>
</ul>
</div>
<div class="keywordblock">
<a name="properties"></a>
<h3>
<span class="kwname">properties:</span>
 <span class="kwallowed">
static_polarizability
</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#symbol_list">(list, more items can be selected)</a></span>
<span class="kwtype"></span>
</h3>
<p></p>
</div>
<div class="keywordblock">
<a name="pseudopotentials"></a>
<h3>
<span class="kwname">pseudopotentials:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Pseudopotential to be applied to all atoms. None used when not set.</p>
</div>
<div class="keywordblock">
<a name="pseudopotentials_elements"></a>
<h3>
<span class="kwname">pseudopotentials_elements:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#hash">hash - a list of name: value pairs</a></span>
</h3>
<p>Specification of pseudopotentials for selected elements. The format is
a hash element: "pseudopotential_name"</p>
</div>
<div class="keywordblock">
<a name="psi4_bin_dir"></a>
<h3>
<span class="kwname">psi4_bin_dir:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Path to the directory containing the psi4 binaries</p>
</div>
<div class="keywordblock">
<a name="psi4_data_dir"></a>
<h3>
<span class="kwname">psi4_data_dir:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Path to the psi4 data directory</p>
</div>
<div class="keywordblock">
<a name="psi4_df_guess"></a>
<h3>
<span class="kwname">psi4_df_guess:</span>
 <span class="kwallowed"><span class="defvalue">yes</span> | no</span>
<span class="kwtype"></span>
</h3>
<p>Use density fitting calculation as a guess for a regular SCF.</p>
</div>
<div class="keywordblock">
<a name="psi4_fno"></a>
<h3>
<span class="kwname">psi4_fno:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Frozen Natural Orbitals approximation in MP3, CCSD and CCSD(T) calculations.</p>
</div>
<div class="keywordblock">
<a name="psi4_geometry_fragments"></a>
<h3>
<span class="kwname">psi4_geometry_fragments:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Separate fragments in DFT geometry. Two fragments should be defined
in blocks molecule_a and molecule_b. Keywords charge, multiplicity and
selection are needed.</p>
</div>
<div class="keywordblock">
<a name="psi4_sapt_ct"></a>
<h3>
<span class="kwname">psi4_sapt_ct:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Switch on additional SAPT calculations to separating charge transfer
energy as a difference between SAPT polarization in monomer and dimer basis.</p>
</div>
<div class="keywordblock">
<a name="psi4_sapt_level"></a>
<h3>
<span class="kwname">psi4_sapt_level:</span>
 <span class="kwallowed">
<span class="defvalue">sapt0</span> | sapt2 | sapt2+ | sapt2+(3) | sapt2+3
</span>
<span class="kwtype"></span>
</h3>
<p></p>
</div>
<div class="keywordblock">
<a name="qmmm_add_ter"></a>
<h3>
<span class="kwname">qmmm_add_ter:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Selection of atoms in QM regieon (selection expression) after which PDB TER labels are added.</p>
</div>
<div class="keywordblock">
<a name="qmmm_auto_fragmentation"></a>
<h3>
<span class="kwname">qmmm_auto_fragmentation:</span>
 <span class="kwallowed">
<span class="defvalue">none</span> | peptide_backbone
</span>
<span class="kwtype"></span>
</h3>
<p>Automated fragmentation of specific systems. When automated fragmentation is
used, the only required input is qmmm_core, input for qmmm_cut_bonds,
qmmm_rename_residues and qmmm_add_ter s generated automatically so that the
QM region can be described by a forcefield modification available in cuby.
Following options are available:</p>
<ul>
<li>none - normal QM/MM run with user-specified input</li>
<li>peptide_backbone - Selects whole amino acids and caps the broken backbone, to be used with amber forcefield defined by amber_leprc: "%interface/data/amberff03_pm6.leaprc"</li>
</ul>
</div>
<div class="keywordblock">
<a name="qmmm_auto_run"></a>
<h3>
<span class="kwname">qmmm_auto_run:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Run the calculation directly after the automated fragmentation. By default,
the generated setup is printed and QM region geometry is saved for manual
inspection.</p>
</div>
<div class="keywordblock">
<a name="qmmm_charges_around_links"></a>
<h3>
<span class="kwname">qmmm_charges_around_links:</span>
 <span class="defvalue">0</span>
<span class="kwtype"></span>
</h3>
<p>Removal of embedding charges around link atoms. The value is a number of bonds
defining the distance. Zero means only the charge on the atom replaced by a
link atom is removed, value of 1 selects also atoms boud to it etc.</p>
</div>
<div class="keywordblock">
<a name="qmmm_charges_extra_info"></a>
<h3>
<span class="kwname">qmmm_charges_extra_info:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Store information on the original atom in the point charges. Not needed
unless a specialized calculation with a QM interface that supports this
is used.</p>
</div>
<div class="keywordblock">
<a name="qmmm_core"></a>
<h3>
<span class="kwname">qmmm_core:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Selection of QM core in QM/MM calculation</p>
</div>
<div class="keywordblock">
<a name="qmmm_cut_bonds"></a>
<h3>
<span class="kwname">qmmm_cut_bonds:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#array">array</a></span>
</h3>
<p>List of bonds that are cut and replaced by link atoms</p>
</div>
<div class="keywordblock">
<a name="qmmm_embedding"></a>
<h3>
<span class="kwname">qmmm_embedding:</span>
 <span class="kwallowed">
<span class="defvalue">mechanical</span> | electrostatic_2way | electrostatic
</span>
<span class="kwtype"></span>
</h3>
<p>The "embedding" used:</p>
<ul>
<li>mechanical - all the intraction between the QM region and the rest of the system is treated by MM</li>
<li>electrostatic_2way - the interaction of the QM region with the point charges modeling the rest of the system is included in the QM Hamiltonian</li>
<li>electrostatic - the same as above but with inexact gradient (energy is the same)</li>
</ul>
</div>
<div class="keywordblock">
<a name="qmmm_geometry_only"></a>
<h3>
<span class="kwname">qmmm_geometry_only:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>If set, the program is aborted after preparation of the QM region but before
the calculations are prepared.</p>
</div>
<div class="keywordblock">
<a name="qmmm_qmregion_file"></a>
<h3>
<span class="kwname">qmmm_qmregion_file:</span>
 <span class="defvalue">"qm_region.pdb"</span>
<span class="kwtype"></span>
</h3>
<p>Name of the file to which geometry is written to. Set to empty string to prevent writting this file.</p>
</div>
<div class="keywordblock">
<a name="qmmm_remove_charges"></a>
<h3>
<span class="kwname">qmmm_remove_charges:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Selection of MM atoms not to be used as point charges in electrostatic embedding.
By default, charges are removed from the atoms in the bonds across QM/MM boundary,
this option can only add to this.</p>
</div>
<div class="keywordblock">
<a name="qmmm_rename_residues"></a>
<h3>
<span class="kwname">qmmm_rename_residues:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#array">array</a></span>
</h3>
<p>List of residues to be renamed in QM region. Format: YAML array of strings "selection new_name".</p>
</div>
<div class="keywordblock">
<a name="qmmm_wat_center"></a>
<h3>
<span class="kwname">qmmm_wat_center:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Selection of atoms defining the center of the solvent sphere</p>
</div>
<div class="keywordblock">
<a name="qmmm_wat_k"></a>
<h3>
<span class="kwname">qmmm_wat_k:</span>
 <span class="defvalue">5.0</span>
<span class="kwtype"></span>
</h3>
<p>Force constant for attraction of the QM water, in kcal/mol/A^2</p>
</div>
<div class="keywordblock">
<a name="qmmm_wat_num"></a>
<h3>
<span class="kwname">qmmm_wat_num:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#float">float</a></span>
<span class="kwtype"></span>
</h3>
<p>Number of waters in the QMMM regions (waters clostest to
the center defined by qmmm_wat_center are selected)</p>
</div>
<div class="keywordblock">
<a name="queue_extra_commands"></a>
<h3>
<span class="kwname">queue_extra_commands:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Additional comands to be inserted into the shell script that executes cuby
on the node - useful for e.g. seting up the environment.</p>
</div>
<div class="keywordblock">
<a name="queue_jobname"></a>
<h3>
<span class="kwname">queue_jobname:</span>
 <span class="defvalue">"cuby4"</span>
<span class="kwtype"></span>
</h3>
<p>Name of the job passed to the queue system.</p>
</div>
<div class="keywordblock">
<a name="queue_name"></a>
<h3>
<span class="kwname">queue_name:</span>
 <span class="defvalue">"N/A"</span>
<span class="kwtype"></span>
</h3>
<p>Name of the queue to be used.</p>
</div>
<div class="keywordblock">
<a name="queue_output_file"></a>
<h3>
<span class="kwname">queue_output_file:</span>
 <span class="defvalue">"LOG"</span>
<span class="kwtype"></span>
</h3>
<p>Name of the file to which output of cuby is redirected.</p>
</div>
<div class="keywordblock">
<a name="queue_parallel"></a>
<h3>
<span class="kwname">queue_parallel:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#integer">integer</a></span>
<span class="kwtype"></span>
</h3>
<p>Number of CPUs allocated in the queue system. If not set, it is calculated from
other keywords as parallel * cuby_threads</p>
</div>
<div class="keywordblock">
<a name="queue_qsub_command"></a>
<h3>
<span class="kwname">queue_qsub_command:</span>
 <span class="defvalue">"qsub"</span>
<span class="kwtype"></span>
</h3>
<p>The command for submitting the jobs to queue.</p>
</div>
<div class="keywordblock">
<a name="queue_qsub_options"></a>
<h3>
<span class="kwname">queue_qsub_options:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Additional commandline options for the command submitting the job (usually named
qsub).</p>
</div>
<div class="keywordblock">
<a name="queue_results_dir"></a>
<h3>
<span class="kwname">queue_results_dir:</span>
 <span class="defvalue">"RESULTS"</span>
<span class="kwtype"></span>
</h3>
<p>Directory where the results are put to after job finishes. Path is relative to the
directory in which the job is submitted.</p>
</div>
<div class="keywordblock">
<a name="queue_scratch_dir"></a>
<h3>
<span class="kwname">queue_scratch_dir:</span>
 <span class="defvalue">"N/A"</span>
<span class="kwtype"></span>
</h3>
<p>Directory where the calculation is run on the node. Evaluated in a bash script, can
contain environment variables. A subdirectory with unique name is created for each job
in the scratch directory.</p>
</div>
<div class="keywordblock">
<a name="queue_submit"></a>
<h3>
<span class="kwname">queue_submit:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Submit the job to queue instead of running it immediately. The submission
system must be configured accordingly.</p>
</div>
<div class="keywordblock">
<a name="queue_system"></a>
<h3>
<span class="kwname">queue_system:</span>
 <span class="kwallowed">
<span class="defvalue">run_in_background</span> | pbs | sge
</span>
<span class="kwtype"></span>
</h3>
<p>Queue system used:</p>
<ul>
<li>run_in_background - example only, runs the cuby job in background immediately.</li>
<li>pbs - use the PBS queue system</li>
<li>sge - use Sun Grid engine</li>
</ul>
</div>
<div class="keywordblock">
<a name="queue_walltime_hrs"></a>
<h3>
<span class="kwname">queue_walltime_hrs:</span>
 <span class="defvalue">24</span>
<span class="kwtype"></span>
</h3>
<p>Requested walltime passed to the queue system.</p>
</div>
<div class="keywordblock">
<a name="random_seed"></a>
<h3>
<span class="kwname">random_seed:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#integer">integer</a></span>
<span class="kwtype"></span>
</h3>
<p>Random seed used to initialize the random number generator when setting up
a MD run. When not set, random initialization is used.</p>
</div>
<div class="keywordblock">
<a name="reaction_formula"></a>
<h3>
<span class="kwname">reaction_formula:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Reaction formula, in format e.g "2A + B -> C + 2D"</p>
</div>
<div class="keywordblock">
<a name="reaction_geometries"></a>
<h3>
<span class="kwname">reaction_geometries:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#hash">hash - a list of name: value pairs</a></span>
</h3>
<p>List of geometries of all molecules, in the order they are presented in the
formula </p>
</div>
<div class="keywordblock">
<a name="reaction_print"></a>
<h3>
<span class="kwname">reaction_print:</span>
 <span class="kwallowed">
<span class="defvalue">formula</span>
</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#symbol_list">(list, more items can be selected)</a></span>
<span class="kwtype"></span>
</h3>
<p>Controls the output of the protocol. Allowed options are:</p>
<ul>
<li>formula - print the reaction formula (on by default)</li>
</ul>
</div>
<div class="keywordblock">
<a name="reaction_smiles"></a>
<h3>
<span class="kwname">reaction_smiles:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Generate the geometries of the molecules using SMILES notation from the
formula itself</p>
</div>
<div class="keywordblock">
<a name="redcoord_all_torsions"></a>
<h3>
<span class="kwname">redcoord_all_torsions:</span>
 <span class="kwallowed"><span class="defvalue">yes</span> | no</span>
<span class="kwtype"></span>
</h3>
<p>Generation of redundant coordinates - use all torsions </p>
</div>
<div class="keywordblock">
<a name="redcoord_extra_contacts"></a>
<h3>
<span class="kwname">redcoord_extra_contacts:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Add extra contacts to the redundant coordinates</p>
</div>
<div class="keywordblock">
<a name="redcoord_freeze_class"></a>
<h3>
<span class="kwname">redcoord_freeze_class:</span>
 <span class="kwallowed">
bonds | angles | torsions
</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#symbol_list">(list, more items can be selected)</a></span>
<span class="kwtype"></span>
</h3>
<p>Freeze some class of internal coordinates</p>
</div>
<div class="keywordblock">
<a name="remote_calculations"></a>
<h3>
<span class="kwname">remote_calculations:</span>
 <span class="kwallowed">
<span class="defvalue">no</span> | drb
</span>
<span class="kwtype"></span>
</h3>
<p>Mode of parallel (specified by -->cuby_threads) remote execution of calcualtions.
The options are:</p>
<ul>
<li>no - calculations are run locally in multiple threads</li>
<li>drb - calculations are run at remote servers using drb (distributed ruby), URIs of the servers must be provided in -->remote_drb_servers</li>
</ul>
</div>
<div class="keywordblock">
<a name="remote_drb_servers"></a>
<h3>
<span class="kwname">remote_drb_servers:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#array">array</a></span>
</h3>
<p>List of URIs of the remote servers (the URI is printed when the server is started)</p>
</div>
<div class="keywordblock">
<a name="remote_file_transfer"></a>
<h3>
<span class="kwname">remote_file_transfer:</span>
 <span class="kwallowed">
shared_wd | scp
</span>
<span class="kwtype"></span>
</h3>
<p>Specifies how the calculation files are transferred to a node:</p>
<ul>
<li>shared_wd - no transfer, all calculations use a shared storage</li>
<li>scp - files are copied using ssh </li>
</ul>
</div>
<div class="keywordblock">
<a name="remote_servers_close"></a>
<h3>
<span class="kwname">remote_servers_close:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Shut down remote servers when the calculation is finished</p>
</div>
<div class="keywordblock">
<a name="remove_rotation"></a>
<h3>
<span class="kwname">remove_rotation:</span>
 <span class="kwallowed"><span class="defvalue">yes</span> | no</span>
<span class="kwtype"></span>
</h3>
<p>Removes rotational component of a gradient in each cycle of
optimization or MD a velocity in MD simulation.</p>
</div>
<div class="keywordblock">
<a name="remove_translation"></a>
<h3>
<span class="kwname">remove_translation:</span>
 <span class="kwallowed"><span class="defvalue">yes</span> | no</span>
<span class="kwtype"></span>
</h3>
<p>Removes translational component of a gradient in each cycle of
optimization or a velocity in MD simulation.</p>
</div>
<div class="keywordblock">
<a name="replace_energy_with_component"></a>
<h3>
<span class="kwname">replace_energy_with_component:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#symbol">symbol</a></span>
<span class="kwtype"></span>
</h3>
<p>Replace the total energy in the result with a selcted component of it. The original energy
is stored in a component "original_energy".</p>
</div>
<div class="keywordblock">
<a name="replace_energy_with_expression"></a>
<h3>
<span class="kwname">replace_energy_with_expression:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Replace the total energy in the result with an arbitrary expression
using the energy components. The original energy is stored in a
component "original_energy".</p>
</div>
<div class="keywordblock">
<a name="restart_file"></a>
<h3>
<span class="kwname">restart_file:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Name of file to which the last geometry is written. When not specified,
the name of the file is "optimized.xyz" if the optimization converged and
"last.xyz" if it did not. Setting the value to "" means no restart file
is written.</p>
</div>
<div class="keywordblock">
<a name="restart_file_append"></a>
<h3>
<span class="kwname">restart_file_append:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>By default, the restart file is overwritten, use this option to append the coordinates
to an existing file</p>
</div>
<div class="keywordblock">
<a name="restraints"></a>
<h3>
<span class="kwname">restraints:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#array">array</a></span>
</h3>
<p>List of restraints. Each line has format:
distance|angle|torsion|distance_difference; value; force_constant; atom_selection1; atom_selection2; ...
Angles are specified in degrees</p>
</div>
<div class="keywordblock">
<a name="s6"></a>
<h3>
<span class="kwname">s6:</span>
 <span class="defvalue">1.0</span>
<span class="kwtype"></span>
</h3>
<p>Scaling of the dispersion energy</p>
</div>
<div class="keywordblock">
<a name="scan_generator"></a>
<h3>
<span class="kwname">scan_generator:</span>
 <span class="kwallowed">
<span class="defvalue">none</span> | grid_xyz
</span>
<span class="kwtype"></span>
</h3>
<p>By default, the scan is performed on a series of geometries provided
by the user. The scan generator can be used to generate this series
on the fly.</p>
</div>
<div class="keywordblock">
<a name="scan_generator_batchsize"></a>
<h3>
<span class="kwname">scan_generator_batchsize:</span>
 <span class="defvalue">0</span>
<span class="kwtype"></span>
</h3>
<p>When set to 0 (default), a series of geometries is saved in a single file and
immediately calculated. If set to nonzero number, the scan is divided into
batches of this size which are saved in separate directories so that they can
be run independently.</p>
</div>
<div class="keywordblock">
<a name="scan_generator_setup"></a>
<h3>
<span class="kwname">scan_generator_setup:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#hash">hash - a list of name: value pairs</a></span>
</h3>
<p>Setup for the scan generator. Different options are available for different
generators, see the protocol documentation.</p>
</div>
<div class="keywordblock">
<a name="scan_mode"></a>
<h3>
<span class="kwname">scan_mode:</span>
 <span class="kwallowed">
serial | <span class="defvalue">parallel</span>
</span>
<span class="kwtype"></span>
</h3>
<p>In the default (parallel) mode, all calculations of teh scan are prepared
and then run using the internal queue system of cuby, which can run
multiple calculations in parallel. In the serial mode, only one
calculation is set up and repeatedly called with the changing geometries.</p>
</div>
<div class="keywordblock">
<a name="scan_print_mode"></a>
<h3>
<span class="kwname">scan_print_mode:</span>
 <span class="kwallowed">
numbers | <span class="defvalue">simple</span> | full
</span>
<span class="kwtype"></span>
</h3>
<p>Mode of printing scan results:</p>
<ul>
<li>numbers - just a column(s) of numbers are printed</li>
<li>simple - one result per line, with some description</li>
<li>full - Verbose output from the child protocol. Only this option is available for all protocols.</li>
</ul>
</div>
<div class="keywordblock">
<a name="scf_convergence"></a>
<h3>
<span class="kwname">scf_convergence:</span>
 <span class="defvalue">7</span>
<span class="kwtype"></span>
</h3>
<p>SCF convergence threshold (energy), set to 10^-X a.u.</p>
</div>
<div class="keywordblock">
<a name="scf_cycles"></a>
<h3>
<span class="kwname">scf_cycles:</span>
 <span class="defvalue">50</span>
<span class="kwtype"></span>
</h3>
<p>Maximum number of SCF iterations</p>
</div>
<div class="keywordblock">
<a name="scf_integral_storage"></a>
<h3>
<span class="kwname">scf_integral_storage:</span>
 <span class="kwallowed">
<span class="defvalue">default</span> | disc | direct | mixed
</span>
<span class="kwtype"></span>
</h3>
<p>Integral storage in SCF. When not set, the default option of
the interfaced program is used. The other options are:</p>
<ul>
<li>disc - store everything on disc</li>
<li>direct - recalculate on the fly</li>
<li>mixed - optin specific to deMon interface</li>
</ul>
</div>
<div class="keywordblock">
<a name="scf_levelshift"></a>
<h3>
<span class="kwname">scf_levelshift:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#float">float</a></span>
<span class="kwtype"></span>
</h3>
<p>Shift of virtual orbitals (in a.u.). When the keyword is not set, default value of the
external program is used.</p>
</div>
<div class="keywordblock">
<a name="scf_multipole_expansion"></a>
<h3>
<span class="kwname">scf_multipole_expansion:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Multipole expansion in evaluation of coulomb integrals.</p>
</div>
<div class="keywordblock">
<a name="scs_os"></a>
<h3>
<span class="kwname">scs_os:</span>
 <span class="defvalue">1.2</span>
<span class="kwtype"></span>
</h3>
<p>SCS scaling for "opposite spin" term. The default value is for SCS-MP2.</p>
</div>
<div class="keywordblock">
<a name="scs_ss"></a>
<h3>
<span class="kwname">scs_ss:</span>
 <span class="defvalue">0.333333</span>
<span class="kwtype"></span>
</h3>
<p>SCS scaling for "same spin" term. The default value is for SCS-MP2.</p>
</div>
<div class="keywordblock">
<a name="selection"></a>
<h3>
<span class="kwname">selection:</span>
 <span class="defvalue">"auto"</span>
<span class="kwtype"></span>
</h3>
<p>Atom selection using cuby's selection language. The default value is
context-dependent:</p>
<ul>
<li>At the root level, where the selection defines what part of the geometry is loaded, the default is %all(), loading everything</li>
<li>In selection of a subsystem (e.g. for interaction energy calculation), first molecule becomes the first subsystem etc...</li>
</ul>
</div>
<div class="keywordblock">
<a name="serial_scf"></a>
<h3>
<span class="kwname">serial_scf:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Run SCF part of calculation in serial and paralellize only the
correlation part.</p>
</div>
<div class="keywordblock">
<a name="shell_commands"></a>
<h3>
<span class="kwname">shell_commands:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Commands to be executed in the default shell (using the system call from
ruby).</p>
</div>
<div class="keywordblock">
<a name="skip_step_if_file_found"></a>
<h3>
<span class="kwname">skip_step_if_file_found:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Condition allowing skipping a step if some file already exists</p>
</div>
<div class="keywordblock">
<a name="skip_step_if_file_not_found"></a>
<h3>
<span class="kwname">skip_step_if_file_not_found:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Condition allowing skipping a step if some file is missing</p>
</div>
<div class="keywordblock">
<a name="solvent_epsilon"></a>
<h3>
<span class="kwname">solvent_epsilon:</span>
 <span class="defvalue">78.5</span>
<span class="kwtype"></span>
</h3>
<p>Dielectric constant of implicit solvent</p>
</div>
<div class="keywordblock">
<a name="solvent_model"></a>
<h3>
<span class="kwname">solvent_model:</span>
 <span class="kwallowed">
<span class="defvalue">none</span> | gbm | igb7 | cosmo
</span>
<span class="kwtype"></span>
</h3>
<p>Implicit solvent model:</p>
<ul>
<li>none</li>
<li>GBM (in AMBER) - Generalized Born Model</li>
<li>IGB7 (in AMBER) - another parameterization of GBM</li>
<li>COSMO (Mopac) - Conductor-like screening model</li>
</ul>
</div>
<div class="keywordblock">
<a name="sos_scaling"></a>
<h3>
<span class="kwname">sos_scaling:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>SOS (scaled oppposite spin) scaling in MP2.
No further input needed, scaling of 1.3 is used.
Nn contrast to full SCS, the second component is not
calculaed at all.
Ref: J. Chem. Phys. 121, 9793 (2004); http://dx.doi.org/10.1063/1.1809602</p>
</div>
<div class="keywordblock">
<a name="sos_scaling_laplace"></a>
<h3>
<span class="kwname">sos_scaling_laplace:</span>
 <span class="defvalue">0</span>
<span class="kwtype"></span>
</h3>
<p>When set to nonzero value, accelerated Laplace-transformed SOS-MP2 is used. 
The value of the keyword specifies the threshold (10^-X a.u.) used 
(4 or 5 is recommended).</p>
</div>
<div class="keywordblock">
<a name="spin_component_scaling"></a>
<h3>
<span class="kwname">spin_component_scaling:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>SCS scaling in MP2. Use scs_ss and scs_ss keywords to set
the scaling coefficients.</p>
</div>
<div class="keywordblock">
<a name="spin_restricted"></a>
<h3>
<span class="kwname">spin_restricted:</span>
 <span class="kwallowed"><span class="defvalue">yes</span> | no</span>
<span class="kwtype"></span>
</h3>
<p>Restricted/Unrestricted spin treatment</p>
</div>
<div class="keywordblock">
<a name="sr"></a>
<h3>
<span class="kwname">sr:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#float">float</a></span>
<span class="kwtype"></span>
</h3>
<p>Radii scaling for damping function</p>
</div>
<div class="keywordblock">
<a name="start_from_previous"></a>
<h3>
<span class="kwname">start_from_previous:</span>
 <span class="kwallowed"><span class="defvalue">yes</span> | no</span>
<span class="kwtype"></span>
</h3>
<p>Use results from previous step (of, e.g. geometry optimization) as
a starting point for the calculation</p>
</div>
<div class="keywordblock">
<a name="step_queue"></a>
<h3>
<span class="kwname">step_queue:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>When set in a step block, the calculation is queued until 
a following steps triggers execution of the queue</p>
</div>
<div class="keywordblock">
<a name="step_title"></a>
<h3>
<span class="kwname">step_title:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Custom step title, step name is used if not set</p>
</div>
<div class="keywordblock">
<a name="step_title_decoration"></a>
<h3>
<span class="kwname">step_title_decoration:</span>
 <span class="defvalue">"="</span>
<span class="kwtype"></span>
</h3>
<p>Character using in the decoration lines in step titles</p>
</div>
<div class="keywordblock">
<a name="steps"></a>
<h3>
<span class="kwname">steps:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#symbol_list">list of symbols</a></span>
<span class="kwtype"></span>
</h3>
<p>List (comma separated, or YAML array) of the names of the steps. Each step is then defined
in a subsection named "calculation_stepname". Common setup shared by all
the steps can be defined in subsection "calculation_common", this setup can
be overriden by the setup of a particular step.</p>
</div>
<div class="keywordblock">
<a name="temperature"></a>
<h3>
<span class="kwname">temperature:</span>
 <span class="defvalue">300.0</span>
<span class="kwtype"></span>
</h3>
<p>Temperature to be maintained by the thermostat</p>
</div>
<div class="keywordblock">
<a name="temperature_target"></a>
<h3>
<span class="kwname">temperature_target:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#float">float</a></span>
<span class="kwtype"></span>
</h3>
<p>When set, the thermostated temperature is varied during the simulation,
starting with the one set by keyword temperature and linearly
interpolating to this value at the end of simulation.</p>
</div>
<div class="keywordblock">
<a name="template_mopac_exe"></a>
<h3>
<span class="kwname">template_mopac_exe:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>MOPAC executable to be used</p>
</div>
<div class="keywordblock">
<a name="test_name"></a>
<h3>
<span class="kwname">test_name:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Name of the test printed in the report</p>
</div>
<div class="keywordblock">
<a name="test_print_results"></a>
<h3>
<span class="kwname">test_print_results:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Print the original results in addition to the test report</p>
</div>
<div class="keywordblock">
<a name="test_result"></a>
<h3>
<span class="kwname">test_result:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>The result the tested calculation should yield. The value is converted
to and tested as quantity defined by keyword test_result_type.</p>
</div>
<div class="keywordblock">
<a name="test_result_type"></a>
<h3>
<span class="kwname">test_result_type:</span>
 <span class="kwallowed">
<span class="defvalue">energy</span> | gradient | dataset
</span>
<span class="kwtype"></span>
</h3>
<p>Type of the result tested:</p>
<ul>
<li>energy - energy is compared</li>
<li>gradient - absolute value of the gradient is compared</li>
<li>dataset - RMSE over the dataset is compared</li>
</ul>
</div>
<div class="keywordblock">
<a name="test_threshold"></a>
<h3>
<span class="kwname">test_threshold:</span>
 <span class="defvalue">0.0001</span>
<span class="kwtype"></span>
</h3>
<p>The threshold defining a successful test.</p>
</div>
<div class="keywordblock">
<a name="thermo_low_mode_fix"></a>
<h3>
<span class="kwname">thermo_low_mode_fix:</span>
 <span class="kwallowed">
<span class="defvalue">none</span> | as_rotations | as_rotations_including_negative
</span>
<span class="kwtype"></span>
</h3>
<p>Special treatment of low frequency modes in thermodynamics calculations:</p>
<ul>
<li>none - no fix applied</li>
<li>as_rotations - Low mode entropy correction by Grimme[DOI: 10.1002/chem.201200497]</li>
<li>as_rotations_including_negative - same as above, including negative frequencies</li>
</ul>
</div>
<div class="keywordblock">
<a name="thermo_pressure"></a>
<h3>
<span class="kwname">thermo_pressure:</span>
 <span class="defvalue">101325.0</span>
<span class="kwtype"></span>
</h3>
<p>Pressure for the thermodynamics calculations, in Pa.
The default is 1 atm.</p>
</div>
<div class="keywordblock">
<a name="thermo_symmetry_number"></a>
<h3>
<span class="kwname">thermo_symmetry_number:</span>
 <span class="defvalue">1.0</span>
<span class="kwtype"></span>
</h3>
<p>Symmetry number for the thermodynamics calculations.</p>
</div>
<div class="keywordblock">
<a name="thermo_temperature"></a>
<h3>
<span class="kwname">thermo_temperature:</span>
 <span class="defvalue">298.15</span>
<span class="kwtype"></span>
</h3>
<p>Temperature for the thermodynamics calculations, in K.</p>
</div>
<div class="keywordblock">
<a name="thermostat"></a>
<h3>
<span class="kwname">thermostat:</span>
 <span class="kwallowed">
<span class="defvalue">none</span> | berendsen | andersen | nose-hoover | bussi
</span>
<span class="kwtype"></span>
</h3>
<p>Selection of thermostat algorithm</p>
</div>
<div class="keywordblock">
<a name="thermostat_tc"></a>
<h3>
<span class="kwname">thermostat_tc:</span>
 <span class="defvalue">0.5</span>
<span class="kwtype"></span>
</h3>
<p>Thermostat time constant (ps) setting strength of the coupling to the thermostat.
The exact mening of the value is different in different algorithms.</p>
</div>
<div class="keywordblock">
<a name="timestep"></a>
<h3>
<span class="kwname">timestep:</span>
 <span class="defvalue">0.001</span>
<span class="kwtype"></span>
</h3>
<p>Timestep for MD (ps)</p>
</div>
<div class="keywordblock">
<a name="trajectory_file"></a>
<h3>
<span class="kwname">trajectory_file:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Name of the file the trajectory is written to. By default, it is
"trajectory" + base name of the input file + a suffix for the selected file format.</p>
</div>
<div class="keywordblock">
<a name="trajectory_format"></a>
<h3>
<span class="kwname">trajectory_format:</span>
 <span class="kwallowed">
<span class="defvalue">xyz_velo</span> | xyz
</span>
<span class="kwtype"></span>
</h3>
<p>Format of the trajectory:</p>
<ul>
<li>xyz_velo - .xyz file with extra columns for velocities in Angstrom/ps units</li>
<li>xyz - plain .xyz format, velocities are not written</li>
</ul>
</div>
<div class="keywordblock">
<a name="trajectory_freq"></a>
<h3>
<span class="kwname">trajectory_freq:</span>
 <span class="defvalue">1</span>
<span class="kwtype"></span>
</h3>
<p>Frequency of writing the trajectory (every Nth step). If set to 0, no
trajectory is written.</p>
</div>
<div class="keywordblock">
<a name="trajectory_selection"></a>
<h3>
<span class="kwname">trajectory_selection:</span>
 <span class="defvalue">"%all()"</span>
<span class="kwtype"></span>
</h3>
<p>Selection of atoms written to the trajectory. By default, trajectory of
the whole system is logged. Optionally, only selected atoms (specifed
using the cuby selection expression) could be saved. A special value
"md_region" makes this selection equal to the selection of the moving
region (see md_region keyword).</p>
</div>
<div class="keywordblock">
<a name="trust_radius"></a>
<h3>
<span class="kwname">trust_radius:</span>
 <span class="defvalue">0.1</span>
<span class="kwtype"></span>
</h3>
<p>Trust radius (max. step size) in Angstrom.</p>
</div>
<div class="keywordblock">
<a name="trust_radius_max"></a>
<h3>
<span class="kwname">trust_radius_max:</span>
 <span class="defvalue">0.3</span>
<span class="kwtype"></span>
</h3>
<p>Upper limit of trust_radius when dynamic change of it is allowed (by opt_tr_update)</p>
</div>
<div class="keywordblock">
<a name="trust_radius_min"></a>
<h3>
<span class="kwname">trust_radius_min:</span>
 <span class="defvalue">1.0e-07</span>
<span class="kwtype"></span>
</h3>
<p>Lower limit of trust_radius when dynamic change of it is allowed (by opt_tr_update)</p>
</div>
<div class="keywordblock">
<a name="turbomole_bin_dir"></a>
<h3>
<span class="kwname">turbomole_bin_dir:</span>
 <span class="defvalue">"N/A"</span>
<span class="kwtype"></span>
</h3>
<p>Path to the turbomole bin directory (serial version)</p>
</div>
<div class="keywordblock">
<a name="turbomole_bin_dir_ga"></a>
<h3>
<span class="kwname">turbomole_bin_dir_ga:</span>
 <span class="defvalue">"N/A"</span>
<span class="kwtype"></span>
</h3>
<p>Path to the turbomole bin directory (global arrays)</p>
</div>
<div class="keywordblock">
<a name="turbomole_bin_dir_mpi"></a>
<h3>
<span class="kwname">turbomole_bin_dir_mpi:</span>
 <span class="defvalue">"N/A"</span>
<span class="kwtype"></span>
</h3>
<p>Path to the turbomole bin directory (mpi)</p>
</div>
<div class="keywordblock">
<a name="turbomole_bin_dir_smp"></a>
<h3>
<span class="kwname">turbomole_bin_dir_smp:</span>
 <span class="defvalue">"N/A"</span>
<span class="kwtype"></span>
</h3>
<p>Path to the turbomole bin directory (shared memory)</p>
</div>
<div class="keywordblock">
<a name="turbomole_keywords"></a>
<h3>
<span class="kwname">turbomole_keywords:</span>
 <span class="defvalue">""</span>
<span class="kwtype"></span>
</h3>
<p>Extra keywords to be added to the turbomole control file.</p>
</div>
<div class="keywordblock">
<a name="turbomole_scf_grad"></a>
<h3>
<span class="kwname">turbomole_scf_grad:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Calculate and save a separate SCF gradient before post-HF gradient is
calculated.</p>
</div>
<div class="keywordblock">
<a name="turbomole_scf_ri_marij"></a>
<h3>
<span class="kwname">turbomole_scf_ri_marij:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Multipole-accelerated resolution of identity approximation for HF/DFT calculation.
Turbomole's default setup is used.</p>
</div>
<div class="keywordblock">
<a name="turbomole_scripts_dir"></a>
<h3>
<span class="kwname">turbomole_scripts_dir:</span>
 <span class="defvalue">"N/A"</span>
<span class="kwtype"></span>
</h3>
<p>Path to the turbomole scripts directory</p>
</div>
<div class="keywordblock">
<a name="turbomole_turbodir"></a>
<h3>
<span class="kwname">turbomole_turbodir:</span>
 <span class="defvalue">"N/A"</span>
<span class="kwtype"></span>
</h3>
<p>Path to the base of turbomole installation</p>
</div>
<div class="keywordblock">
<a name="turbomole_version"></a>
<h3>
<span class="kwname">turbomole_version:</span>
 <span class="defvalue">N/A</span>
<span class="kwtype"></span>
</h3>
<p>Version of turbomole used</p>
</div>
<div class="keywordblock">
<a name="use_symmetry"></a>
<h3>
<span class="kwname">use_symmetry:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>Allow the use of symmetry in the QM calculation</p>
</div>
<div class="keywordblock">
<a name="velocities"></a>
<h3>
<span class="kwname">velocities:</span>
 <span class="kwallowed">
<span class="defvalue">random</span> | zero | read
</span>
<span class="kwtype"></span>
</h3>
<p>Assignment of initial velocities. The options are:</p>
<ul>
<li>random - random velocities from a Gaussian distribution coresponding to the temperature set by init_temp keyword</li>
<li>zero - simulation starts with zero velocities</li>
<li>read - read velocities from a file specified by the velocities_file keyword</li>
</ul>
</div>
<div class="keywordblock">
<a name="velocities_file"></a>
<h3>
<span class="kwname">velocities_file:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p></p>
</div>
<div class="keywordblock">
<a name="verbosity"></a>
<h3>
<span class="kwname">verbosity:</span>
 <span class="kwallowed">
silent | minimal | brief | <span class="defvalue">normal</span> | debug
</span>
<span class="kwtype"></span>
</h3>
<p>Verbosity of the output:</p>
<ul>
<li>silent - nothing is printed</li>
<li>minimal - just the most important results are printed in format intended for machine reading</li>
<li>brief - just the most important results are printed in human readable form</li>
<li>normal - all necessary iinformation is printed with 'decorations'</li>
<li>debug - same as normal + debugging messages</li>
</ul>
</div>
<div class="keywordblock">
<a name="void_delay"></a>
<h3>
<span class="kwname">void_delay:</span>
 <span class="defvalue">0.0</span>
<span class="kwtype"></span>
</h3>
<p>Time (in seconds) spent in the void "calculation"</p>
</div>
<div class="keywordblock">
<a name="waterball_center"></a>
<h3>
<span class="kwname">waterball_center:</span>
 <span class="kwallowed">
solute | origin | fixed_solute
</span>
<span class="kwtype"></span>
</h3>
<p>Center of the waterball:</p>
<ul>
<li>solute - waterball defined relative to the solute center of mass</li>
<li>origin - 0,0,0 coordinate</li>
<li>fixed_solute - 0,0,0 coordinate for waterball, the solute center of mass is connected to it by a harmonic restraint</li>
</ul>
</div>
<div class="keywordblock">
<a name="waterball_solute"></a>
<h3>
<span class="kwname">waterball_solute:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Selection defining the solute or the center of solvation in it. Required when
waterball_center is set to solute or fixed_solute</p>
</div>
<div class="keywordblock">
<a name="waterball_solute_k"></a>
<h3>
<span class="kwname">waterball_solute_k:</span>
 <span class="defvalue">5.0</span>
<span class="kwtype"></span>
</h3>
<p>Force constant of the restraint holding the solute close to the origin
of the coordinate system, applies only to waterball_center = fixed_solute </p>
</div>
<div class="keywordblock">
<a name="write_parsed_input"></a>
<h3>
<span class="kwname">write_parsed_input:</span>
 <span class="kwallowed">yes | <span class="defvalue">no</span></span>
<span class="kwtype"></span>
</h3>
<p>If enabled, the parsed input, including all modifications by cuby
(e.g. child calculation setup and defaults used) is saved to file
"parsed_input.yaml".</p>
</div>
<div class="keywordblock">
<a name="write_results_yaml"></a>
<h3>
<span class="kwname">write_results_yaml:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#string">string</a></span>
<span class="kwtype"></span>
</h3>
<p>Write serialized results in YAML format to this file.</p>
</div>
<div class="keywordblock">
<a name="x0"></a>
<h3>
<span class="kwname">x0:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#float">float</a></span>
<span class="kwtype"></span>
</h3>
<p>A generic parameter used in development</p>
</div>
<div class="keywordblock">
<a name="x1"></a>
<h3>
<span class="kwname">x1:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#float">float</a></span>
<span class="kwtype"></span>
</h3>
<p>A generic parameter used in development</p>
</div>
<div class="keywordblock">
<a name="x2"></a>
<h3>
<span class="kwname">x2:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#float">float</a></span>
<span class="kwtype"></span>
</h3>
<p>A generic parameter used in development</p>
</div>
<div class="keywordblock">
<a name="x3"></a>
<h3>
<span class="kwname">x3:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#float">float</a></span>
<span class="kwtype"></span>
</h3>
<p>A generic parameter used in development</p>
</div>
<div class="keywordblock">
<a name="x4"></a>
<h3>
<span class="kwname">x4:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#float">float</a></span>
<span class="kwtype"></span>
</h3>
<p>A generic parameter used in development</p>
</div>
<div class="keywordblock">
<a name="x5"></a>
<h3>
<span class="kwname">x5:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#float">float</a></span>
<span class="kwtype"></span>
</h3>
<p>A generic parameter used in development</p>
</div>
<div class="keywordblock">
<a name="x6"></a>
<h3>
<span class="kwname">x6:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#float">float</a></span>
<span class="kwtype"></span>
</h3>
<p>A generic parameter used in development</p>
</div>
<div class="keywordblock">
<a name="x7"></a>
<h3>
<span class="kwname">x7:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#float">float</a></span>
<span class="kwtype"></span>
</h3>
<p>A generic parameter used in development</p>
</div>
<div class="keywordblock">
<a name="x8"></a>
<h3>
<span class="kwname">x8:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#float">float</a></span>
<span class="kwtype"></span>
</h3>
<p>A generic parameter used in development</p>
</div>
<div class="keywordblock">
<a name="x9"></a>
<h3>
<span class="kwname">x9:</span>
 <span class="kwtype"><a class="link_type" href="input_types.html#float">float</a></span>
<span class="kwtype"></span>
</h3>
<p>A generic parameter used in development</p>
</div>
<div class="keywordblock">
<a name="xbond_only_noncovalent"></a>
<h3>
<span class="kwname">xbond_only_noncovalent:</span>
 <span class="kwallowed"><span class="defvalue">yes</span> | no</span>
<span class="kwtype"></span>
</h3>
<p>When set, covalent bonds and 1-3 and 1-4 covament neigbors are excluded
from the corrected pairs.</p>
</div>
<div class="keywordblock">
<a name="xbond_parameters"></a>
<h3>
<span class="kwname">xbond_parameters:</span>
<span class="kwtype"><a class="link_type" href="input_types.html#hash">hash - a list of name: value pairs</a></span>
</h3>
<p>Two-level hash, indexed by 1) halogen element and 2) acceptor element, the
value should be array of the k and e parameters. The parameters defined here
are merged with the default parameters for D3H4X correction.</p>
</div>
		</div>
		</div>
	</body>
</html>
